<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="The phruta R package is designed to simplify the basic
            phylogenetic pipeline. Specifically, all code is run within
            the same program and data from intermediate steps are saved in
            independent folders. Furthrmore, all code is run within the
            same environment which increases the reproducibility of your analysis.
            phruta retrieves gene sequences, combines newly downloaded and local
            gene sequences, and performs sequence alignments.">
<title>Phylogenetic Reconstruction and Time-dating • phruta</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Phylogenetic Reconstruction and Time-dating">
<meta property="og:description" content="The phruta R package is designed to simplify the basic
            phylogenetic pipeline. Specifically, all code is run within
            the same program and data from intermediate steps are saved in
            independent folders. Furthrmore, all code is run within the
            same environment which increases the reproducibility of your analysis.
            phruta retrieves gene sequences, combines newly downloaded and local
            gene sequences, and performs sequence alignments.">
<meta property="og:image" content="https://ropensci.github.io/phruta/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">phruta</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="articles/phruta.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="articles/index.html">Articles</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="http://twitter.com/cromanpa" aria-label="Twitter">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ropensci/phruta/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><p><!-- badges: start --> <a href="https://codecov.io/gh/ropensci/phruta?branch=main" class="external-link"><img src="https://codecov.io/gh/ropensci/phruta/branch/main/graph/badge.svg" alt="Codecov test coverage"></a> <a href="https://github.com/ropensci/phruta/actions" class="external-link"><img src="https://github.com/ropensci/phruta/workflows/R-CMD-check/badge.svg" alt="R-CMD-check"></a> <a href="https://lifecycle.r-lib.org/articles/stages.html#maturing" class="external-link"><img src="https://img.shields.io/badge/lifecycle-maturing-blue.svg"></a><br><a href="https://github.com/ropensci/phruta" class="external-link"><img src="https://img.shields.io/github/languages/code-size/ropensci/phruta.svg"></a> <a href="https://www.codefactor.io/repository/github/ropensci/phruta" class="external-link"><img src="https://www.codefactor.io/repository/github/ropensci/phruta/badge" alt="CodeFactor"></a><br><a href="https://github.com/ropensci/software-review/issues/458" class="external-link"><img src="https://badges.ropensci.org/458_status.svg" alt="Status at rOpenSci Software Peer Review"></a> <!-- badges: end --></p>
<div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="the-phruta-r-package-">The <code>phruta</code> <code>R</code> package 
<a class="anchor" aria-label="anchor" href="#the-phruta-r-package-"></a>
</h1>
</div>
<div class="section level3">
<h3 id="assembling-phylogenetic-trees-from-taxonomic-names">Assembling phylogenetic trees from taxonomic names<a class="anchor" aria-label="anchor" href="#assembling-phylogenetic-trees-from-taxonomic-names"></a>
</h3>
</div>
<div class="section level2">
<h2 id="what-is-phruta">What is <code>phruta</code>?<a class="anchor" aria-label="anchor" href="#what-is-phruta"></a>
</h2>
<p>The <code>phruta</code> R package is designed to simplify the basic phylogenetic pipeline. All the code is run within the same program and data from intermediate steps are saved in independent folders (optional). <code>phruta</code> retrieves gene sequences, combines newly downloaded to local gene sequences, performs sequence alignments, and basic phylogenetic inference.</p>
</div>
<div class="section level2">
<h2 id="who-should-consider-using-phruta">Who should consider using <code>phruta</code>
<a class="anchor" aria-label="anchor" href="#who-should-consider-using-phruta"></a>
</h2>
<p>The main functions in the <code>phruta</code> R package allow for a quick mining and curation of GenBank sequences. This package is designed for students and researchers interested in generating species-level genetic datasets for particular sets of taxa. Specifically, if you have a clade or group of species in mind, <code>phruta</code> will help you to assemble a molecular dataset with information available in GenBank.</p>
</div>
<div class="section level2">
<h2 id="why-use-phruta">Why use <code>phruta</code>?<a class="anchor" aria-label="anchor" href="#why-use-phruta"></a>
</h2>
<p><code>phruta</code> simplifies the phylogenetic pipeline, increases reproducibility, and helps organizing information used to infer molecular phylogenies.</p>
</div>
<div class="section level2">
<h2 id="how-is-phruta-different-from-other-software">How is <code>phruta</code> different from other software?<a class="anchor" aria-label="anchor" href="#how-is-phruta-different-from-other-software"></a>
</h2>
<p><code>phruta</code> has two core functions. The main applications of these functions is briefly outlined below:</p>
<ul>
<li><p><code><a href="reference/sq.retrieve.direct.html">sq.retrieve.direct()</a></code> and <code><a href="reference/sq.retrieve.indirect.html">sq.retrieve.indirect()</a></code>: These functions downloads sequences from genbank (nucleotide database) for particular taxa (taxonomic groups or particular species) and a list of genes.</p></li>
<li><p><code><a href="reference/sq.curate.html">sq.curate()</a></code>: After sequences are downloaded from genbank, this function curates sequences within each of the examined genes by detecting sequence outliers and by using taxonomic information.</p></li>
</ul>
<p>In addition to these two main functions, users will be able to align the downloaded sequences, infer phylogenetic trees, and calibrate phylogenies using additional functions in <code>phruta</code>.</p>
</div>
<div class="section level2">
<h2 id="installing-phruta">Installing <code>phruta</code>
<a class="anchor" aria-label="anchor" href="#installing-phruta"></a>
</h2>
<p><code>phruta</code> is currently only available through <code>GitHub</code>. It can be easily installed using the following code.</p>
<pre><code><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span> 
<span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"ropensci/phruta"</span><span class="op">)</span></code></pre>
<p>Alternatively, you can install <code>phruta</code> using:</p>
<pre><code><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"phruta"</span>, repos <span class="op">=</span> <span class="st">"https://ropensci.r-universe.dev"</span><span class="op">)</span></code></pre>
<p>Please make sure that the <code>R</code> packages <code>msa</code>, <code>DECIPHER</code>, <code>Biostrings</code>, and <code>odseq</code> are correctly installed. If you are interested in using the development version of <code>phruta</code>, please install it using the following code:</p>
<pre><code><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span>
<span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"ropensci/phruta"</span>, ref <span class="op">=</span> <span class="st">"dev"</span><span class="op">)</span></code></pre>
</div>
<div class="section level2">
<h2 id="running-phruta-from-shiny">Running <code>phruta</code> from shiny<a class="anchor" aria-label="anchor" href="#running-phruta-from-shiny"></a>
</h2>
<p>I have constructed a shiny app that hosts <code>phruta</code> and enables users to run the basic functions in a less-code intensive environment. The app, <code>salphycon</code> is currently available in the following <a href="https://github.com/cromanpa94/salphycon" class="external-link">GitHub repo</a>. The shiny app will be live at some point in 2023.</p>
</div>
<div class="section level2">
<h2 id="installing-raxml-">Installing RAxML <a name="paragraph1"></a>
<a class="anchor" aria-label="anchor" href="#installing-raxml-"></a>
</h2>
<p>In <code>MacOS</code>, <code>RAxML</code> can be easily installed to the <code>PATH</code> using one of the two lines below in <code>conda</code>:</p>
<p><code>{bash eval=FALSE} conda install -c bioconda/label/cf201901 raxml</code></p>
<p><code>{bash eval=FALSE} conda install -c bioconda raxml</code></p>
<p>For other <code>OS</code> (Windows, Linux), please follow the instructions listed in the official <code>RAxML</code> <a href="https://cme.h-its.org/exelixis/web/software/raxml/" class="external-link">website</a></p>
<p>Once <code>RAxML</code> has been installed to your computer, open <code>R</code> and make sure that the following line doesn’t throw an error.</p>
<p><code>{r eval=FALSE} system("raxmlHPC")</code></p>
<p>Depending on how <code>RAxML</code> was installed, you may want to check if <code>RAxML</code> is called from the terminal using <code>raxmlHPC</code> or <code>raxmlHPC</code>. This string needs to be passed to <code>tree.raxml</code> using the argument <code>raxml_exec</code>. Please note that this argument corresponds to the <code>exec</code> argument in <code><a href="https://rdrr.io/pkg/ips/man/raxml.html" class="external-link">ips::raxml</a></code>.</p>
<p>Finally, note that <code>RStudio</code> sometimes has issues finding <em>stuff</em> in the path while using <code><a href="https://rdrr.io/r/base/system.html" class="external-link">system()</a></code>. If you’re using <code>macOS</code>, try starting <code>RStudio</code> from the command line by running the following line:</p>
<p><code>{bash eval=FALSE} open /Applications/RStudio.app</code></p>
<p>VS code does not suffer of the same issues. In other OS, it might be better to simply avoid using <code>RStudio</code> if you’re interested in running the phylogenetic functions in <code>phruta</code>.</p>
</div>
<div class="section level2">
<h2 id="installing-pathd-8-and-treepl-">Installing <code>PATHd-8</code> and <code>treePL</code> <a name="paragraph2"></a>
<a class="anchor" aria-label="anchor" href="#installing-pathd-8-and-treepl-"></a>
</h2>
<p>There are excellent guides for installing <code>PATHd-8</code> and <code>treePL</code>. Here, I summarize two potentially relevant options.</p>
<p>First, you can use <a href="https://github.com/bomeara/phydocker/blob/master/Dockerfile" class="external-link">Brian O’Meara’s</a> approach for installing <code>PATHd-8</code> in MacOs and linux. I summarize the code in the following <a href="https://gist.github.com/cromanpa94/a43bc710a17220f71d796d6590ea7fe4" class="external-link">link</a>. For Windows users, please use the compiled version of the software provided in the following <a href="https://www2.math.su.se/PATHd8/" class="external-link">link</a>.</p>
<p>Second, you can use homebrew to install <code>treePL</code> (Windows, MacOS, and Linux), thanks to Jonathan Chang.</p>
<p><code>{bash eval = F} brew install brewsci/bio/treepl</code></p>
<p>Please check the following <a href="https://docs.brew.sh/Homebrew-on-Linux" class="external-link">link</a>) if you’re interested in running <code>brew</code> from Windows and Linux.</p>
</div>
<div class="section level2">
<h2 id="running-phruta-from-rstudio-while-using-macos">Running <code>phruta</code> from <code>Rstudio</code> while using <code>MacOS</code>?<a class="anchor" aria-label="anchor" href="#running-phruta-from-rstudio-while-using-macos"></a>
</h2>
<p>Only if you’re interested in running phylogenetic analyses, please make sure you open <code>RStudio</code> using the following code from the terminal:</p>
<p><code>{bash eval=FALSE} open /Applications/RStudio.app</code></p>
</div>
<div class="section level2">
<h2 id="dedication">Dedication<a class="anchor" aria-label="anchor" href="#dedication"></a>
</h2>
<p>My package is dedicated to my mom. I still have lots of things to learn from you. You will always have all my admiration. The <a href="https://www.flickr.com/photos/gufomusike/3462117620/in/photolist-6NFiPi-xoLbca-FtC6yJ-4nk6wS-x2AZV-b3MUv8-e2B7qj-4uCwwa-e3PJxi-2ePGmUM-b2wBVi-obHf1x-5iP26P-4juoE6-z881E-z88t3-9GmTbQ-dGvrFe-22APdBs-p2t5Zv-8DWQw8-6fAJ2G-7jQhu2-7LEkkL-7vBdyF-jTdXSR-kcntD1-aWGfnx-bk59CK-5JfhKt-6gWfX7-reVehy-bjk7Ki-2xnGjv-dLJbq9-e3VjY3-ugz6U-FGVagm-iqVRuD-YE5pLe-2kPkt84-2kHhswd" class="external-link">logo</a> features a Palenquera in Cartagena (Colombia). For many folks, Palenqueras are just the Black woman ones who sell fruits in particular Colombian turistic areas. However, palenqueras and Palenque are central to Black identity in Colombia, Latin America, and across the America: “Palenque was the first free African town in the Americas”](<a href="https://en.wikipedia.org/wiki/San_Basilio_de_Palenque" class="external-link uri">https://en.wikipedia.org/wiki/San_Basilio_de_Palenque</a>).</p>
</div>
<div class="section level2">
<h2 id="etymology">Etymology<a class="anchor" aria-label="anchor" href="#etymology"></a>
</h2>
<p><em>Fruta</em> is the Spanish word for <em>Fruit</em>. English <em>ph</em> sounds the same as <em>F</em> in Spanish. In <code>phruta</code>, <em>ph</em> is relative to phylogenetics. I pronounce <code>phruta</code> just as <em>fruta</em> in Spanish.</p>
</div>
<div class="section level2">
<h2 id="additional-resources">Additional resources<a class="anchor" aria-label="anchor" href="#additional-resources"></a>
</h2>
<p>More details about the functions implemented in <code>phruta</code> can be found in the different vignettes associated with the package or in our <a href="https://ropensci.github.io/phruta/">website</a>.</p>
</div>
<div class="section level2">
<h2 id="alternatives-to-phruta">Alternatives to <code>phruta</code>
<a class="anchor" aria-label="anchor" href="#alternatives-to-phruta"></a>
</h2>
<p>Similar functionalities for assembling curated molecular datasets for phylogenetic analyses can be found in <a href="https://github.com/ropensci/phylotaR" class="external-link"><code>phylotaR</code></a> and <a href="https://github.com/dportik/SuperCRUNCH" class="external-link">SuperCRUNCH</a>. However, note that <code>phylotaR</code> is limited to downloading and curating sequences (e.g. doesn’t align sequences). Similarly, <code>SuperCRUNCH</code> only curates local sequences. <code>phruta</code> is closer to the <a href="https://academic.oup.com/sysbio/article/66/2/152/2418028" class="external-link"><code>SUPERSMART</code></a> and its “new” associated <code>R</code> workflow <a href="https://github.com/AntonelliLab/supersmartR" class="external-link"><code>SUPERSMARTR</code></a>. However, most of the applications in the different packages that are part of <code>SUPERSMARTR</code> are simplified in <code>phruta</code>.</p>
</div>
<div class="section level2">
<h2 id="contributing">Contributing<a class="anchor" aria-label="anchor" href="#contributing"></a>
</h2>
<p>Please see our <a href="CONTRIBUTING">contributing guide</a>.</p>
</div>
<div class="section level2">
<h2 id="contact">Contact<a class="anchor" aria-label="anchor" href="#contact"></a>
</h2>
<p>Please see the package <a href="DESCRIPTION">DESCRIPTION</a> for package authors.</p>
</div>
<div class="section level2">
<h2 id="code-of-conduct">Code of conduct<a class="anchor" aria-label="anchor" href="#code-of-conduct"></a>
</h2>
<p>Please note that this package is released with a <a href="https://ropensci.org/code-of-conduct/" class="external-link">Contributor Code of Conduct</a>. By contributing to this project, you agree to abide by its terms.</p>
</div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/ropensci/phruta/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/ropensci/phruta/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing phruta</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="http://ropensci.github.io/cromanpa/" class="external-link">Cristian Roman Palacios</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0003-1696-4886" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="http://ropensci.github.io/cromanpa/" class="external-link">Cristian Roman Palacios</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
