<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="phruta">
<title>To export or not export phruta outputs ‚Ä¢ phruta</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="To export or not export phruta outputs">
<meta property="og:description" content="phruta">
<meta property="og:image" content="https://cromanpa94.github.io/phruta/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">phruta</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/phruta.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../articles/index.html">Articles</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="http://twitter.com/cromanpa" aria-label="Twitter">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/cromanpa94/phruta/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>To export or not export phruta outputs</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/cromanpa94/phruta/blob/HEAD/vignettes/Exporting_data_phruta.Rmd" class="external-link"><code>vignettes/Exporting_data_phruta.Rmd</code></a></small>
      <div class="d-none name"><code>Exporting_data_phruta.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="table-of-contents">Table of Contents<a class="anchor" aria-label="anchor" href="#table-of-contents"></a>
</h2>
<ol style="list-style-type: decimal">
<li><a href="#intro">To export or not export from
<code>phruta</code></a></li>
<li><a href="#paragraph1">From taxonomic names to sequence alignments,
exporting</a></li>
</ol>
<div class="section level3">
<h3 id="to-export-or-not-export-from-phruta">To export or not export from <code>phruta</code>
<a name="intro"></a><a class="anchor" aria-label="anchor" href="#to-export-or-not-export-from-phruta"></a>
</h3>
<p>Assembling a molecular dataset for particular target taxa with
<code>phruta</code> can be performed almost entirely by saving objects
into your workspace. This topic was covered in the introductory vignette
to <code>phruta</code> (Using the <code>phruta</code> R package).
However, in some situations, it might be desirable to save the outputs
of different <code>phruta</code> functions to particular folders. This
tutorial will cover that specific situation. Specifically, we will be
reviewing how <code>phruta</code> can be used to export
<code>.fasta</code> and <code>.csv</code> files that are generated in
different steps of the pipeline.</p>
</div>
<div class="section level3">
<h3 id="from-taxonomic-names-to-sequence-alignments-exporting-data">From taxonomic names to sequence alignments, exporting data
<a name="paragraph1"></a><a class="anchor" aria-label="anchor" href="#from-taxonomic-names-to-sequence-alignments-exporting-data"></a>
</h3>
<p>In the introductory vignette to <code>phruta</code> (Using the
<code>phruta</code> R package), we assembled a basic molecular dataset
for inferring the phylogeny among three mammal genera. Let‚Äôs recreate
the same tutorial, but in this case, exporting the intermediate files
that are created after using several of the functions. Note that the
structure of this tutorial closely follows that of ‚ÄúUsing the
<code>phruta</code> <code>R</code> package‚Äù. You should be able to
follow this tutorial even without having reviewed the introductory
vignette.</p>
<p>Please assume that we are interested in building a phylogenetic tree
for the following three genera: <em>Felis, Vulpes</em>, and
<em>Phoca</em>. All these three genera are classified within the
Carnivora, a mammalian order. Both <em>Felis</em> and <em>Vulpes</em>
are classified in different superfamilies within the Fissipedia.
Finally, <em>Phoca</em> is part of another suborder, Pinnipedia. We‚Äôre
going to root our tree with another mammal species, a Chinese Pangolin
(<em>Manis pentadactyla</em>). Users can select additional target
species and clades. However, for simplicity, we will run the analyses
using three genera in the ingroup and a single outgroup species.</p>
<p>So far, we have decided the taxonomic make of our analyses in
<code>phruta</code>. We will also need to determine the gene regions to
be used in our analyses. Fortunately, mammals are extensively studied
and a comprehensive list of potential gene regions to be analyzed is
already available. For instance, we could use same gene regions sampled
in <a href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.3000494#sec030" class="external-link">Upham
et al (2009)</a>. However, for this tutorial, we will simply try to find
the gene regions are well sampled for the target taxa. I believe that
figuring out the best sampled gene regions in genbank, instead of
providing gene names, is potentially more valuable when working with
poorly studied groups (e.g.¬†invertebrates). Before we move on, please
make sure that you you have <a href="http://rfunction.com/archives/1001" class="external-link">set a working directory for
this project</a>. All the files will be saved to this directory.
<code>phruta</code> will politely ask before writing files to your local
directories.</p>
<p>Let‚Äôs start by loading <code>phruta</code>!</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cromanpa94/phruta" class="external-link">phruta</a></span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Registered S3 methods overwritten by 'TreeDist':</span>
<span class="co">##   method                 from     </span>
<span class="co">##   as.matrix.ClusterTable TreeTools</span>
<span class="co">##   print.ClusterTable     TreeTools</span>
<span class="co">##   summary.ClusterTable   TreeTools</span></code></pre>
<pre><code><span class="co">##        ‚úÖ  Thank you for using the {phruta} R package!   ‚úÖ</span>
<span class="co">## </span>
<span class="co">##                         üôÇHappy coding!!üôÇ</span>
<span class="co">##                        üçäüççüçêü•ùü••üå∂ü•ëüçíü•≠</span></code></pre>
<p>Now, let‚Äôs look for the gene regions that are sampled for our target
taxa. Again, this step is not always necessary. In some groups,
gene-level sampling is very standard (e.g.¬†COI, 12S). However, the
structure of gene sampling sometimes becomes more blurry as you zoom out
taxonomically. For instance, genes A and B can be extensively sampled in
genus 1. However, genus 2 in the same family has mainly been studied
using genes Y and Z. The idea here is that <code>phruta</code> will try
to find those gene regions that are extensively sampled across species
in the target taxa. We will use the
<code><a href="../reference/gene.sampling.retrieve.html">gene.sampling.retrieve()</a></code> function in <code>phruta</code>.
The resulting <code>data.frame</code>, named <code>gs.seqs</code> in
this example, will contain the list of full names for genes sampled in
genbank for the target taxa.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gs.seqs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gene.sampling.retrieve.html">gene.sampling.retrieve</a></span><span class="op">(</span>organism <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Felis"</span>, <span class="st">"Vulpes"</span>, <span class="st">"Phoca"</span>, <span class="st">"Manis_pentadactyla"</span><span class="op">)</span>, 
                                  speciesSampling <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>For the search terms used above, <code>phruta</code> was able to
retrieve the names for 2133 gene regions. In the table below I summarize
a few of those genes, with sampling frequency calculated at the level of
species (see <code>speciesSampling = TRUE</code> argument above).</p>
<table class="table">
<caption>Gene sampling for the search including the genera Felis,
Vulpes, Phoca as the ingroup taxa. Manis pentadactyla was used as
outgroup.</caption>
<colgroup>
<col width="40%">
<col width="27%">
<col width="31%">
</colgroup>
<thead><tr class="header">
<th align="left">Gene</th>
<th align="right">Sampled in N species</th>
<th align="right">PercentOfSampledSpecies</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">cytochrome b</td>
<td align="right">23</td>
<td align="right">74.19355</td>
</tr>
<tr class="even">
<td align="left">NADH dehydrogenase subunit 5</td>
<td align="right">13</td>
<td align="right">41.93548</td>
</tr>
<tr class="odd">
<td align="left">cytochrome c oxidase subunit I</td>
<td align="right">12</td>
<td align="right">38.70968</td>
</tr>
<tr class="even">
<td align="left">16S ribosomal RNA</td>
<td align="right">11</td>
<td align="right">35.48387</td>
</tr>
<tr class="odd">
<td align="left">12S ribosomal RNA</td>
<td align="right">10</td>
<td align="right">32.25806</td>
</tr>
<tr class="even">
<td align="left">apolipoprotein B</td>
<td align="right">9</td>
<td align="right">29.03226</td>
</tr>
</tbody>
</table>
<p>Thus, the <code><a href="../reference/gene.sampling.retrieve.html">gene.sampling.retrieve()</a></code> function provides an
estimate of the number of species in genbank that matches the taxonomic
criteria and have sequences for a given gene region. Note that the
estimates recovered by <code><a href="../reference/gene.sampling.retrieve.html">gene.sampling.retrieve()</a></code> are only as
good as the annotations that other researchers have provided for
sequences deposited in genbank.</p>
<p>From here, we will generate a preliminary table summarizing accession
numbers for the combination of taxa and gene regions that we‚Äôre
interested in sampling. However, note that not all these accession
numbers are expected to be in the final (curated) molecular dataset. For
instance, several sequences might be dropped later after taxonomic
information is curated. Now, we will assemble a species-level summary of
accession numbers using the <code><a href="../reference/acc.table.retrieve.html">acc.table.retrieve()</a></code> function.
For simplicity, this tutorial will focus on sampling gene regions that
are sampled in &gt;30% of the species (<code>targetGenes</code>
<code>data.frame</code>).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">targetGenes</span> <span class="op">&lt;-</span> <span class="va">gs.seqs</span><span class="op">[</span><span class="va">gs.seqs</span><span class="op">$</span><span class="va">PercentOfSampledSpecies</span> <span class="op">&gt;</span> <span class="fl">30</span>,<span class="op">]</span>

<span class="va">acc.table</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/acc.table.retrieve.html">acc.table.retrieve</a></span><span class="op">(</span>
            clades  <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Felis'</span>, <span class="st">'Vulpes'</span>, <span class="st">'Phoca'</span><span class="op">)</span>,
            species <span class="op">=</span> <span class="st">'Manis_pentadactyla'</span> ,
            genes   <span class="op">=</span> <span class="va">targetGenes</span><span class="op">$</span><span class="va">Gene</span>,
            speciesLevel <span class="op">=</span> <span class="cn">TRUE</span>
          <span class="op">)</span></code></pre></div>
<p>The <code>acc.table</code> object is a <code>data.frame</code> that
will be used below for downloading the relevant gene sequences. In this
case, the dataset includes the following information:</p>
<table class="table">
<caption>Accession numbers for the relevant species and gene regions
examined in this tutorial.</caption>
<colgroup>
<col width="5%">
<col width="87%">
<col width="1%">
<col width="5%">
</colgroup>
<thead><tr class="header">
<th align="left">Species</th>
<th align="left">Ti</th>
<th align="left">Acc</th>
<th align="left">gene</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Felis catus</td>
<td align="left">Felis catus MKRaS006 mitochondrial gene for cytochrome
b, partial cds</td>
<td align="left">LC649704</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Felis silvestris</td>
<td align="left">Felis silvestris lybica isolate CAT157 NADH
dehydrogenase subunit 5 (ND5) gene, partial cds; NADH dehydrogenase
subunit 6 (ND6) gene, complete cds; tRNA-Glu (trnE(ttc)) gene, complete
sequence; and cytochrome b (cytb) gene, partial cds; mitochondrial</td>
<td align="left">MN537982</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Felis chaus</td>
<td align="left">Felis chaus isolate Jungle Cat 5 cytochrome b (cytb)
gene, partial cds; mitochondrial</td>
<td align="left">MN370575</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Felis environmental</td>
<td align="left">Felis environmental sample isolate Kw-170 cytochrome b
(cytb) gene, partial cds; mitochondrial</td>
<td align="left">MK510873</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Felis margarita</td>
<td align="left">Felis margarita haplotype AH NADH dehydrogenase subunit
5 (ND5) and cytochrome b (cytb) genes, partial cds; and tRNA-Thr gene
and D-loop, partial sequence; mitochondrial</td>
<td align="left">MK606132</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Felis bieti</td>
<td align="left">Felis bieti cytochrome b gene, partial cds;
mitochondrial</td>
<td align="left">AY773081</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">F.domesticus mitochondrial</td>
<td align="left">F.domesticus mitochondrial cytochrome b gene</td>
<td align="left">X82296</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Vulpes corsac</td>
<td align="left">Vulpes corsac isolate SH21 cytochrome b (CYTB) gene,
complete cds; mitochondrial</td>
<td align="left">MT795179</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Vulpes vulpes</td>
<td align="left">Vulpes vulpes isolate Kharkiv10 cytochrome b (cytb)
gene, complete cds; mitochondrial</td>
<td align="left">EF689062</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Vulpes zerda</td>
<td align="left">Vulpes zerda isolate X161349 cytochrome b (Cytb) gene,
partial cds; mitochondrial</td>
<td align="left">MH854561</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Vulpes cana</td>
<td align="left">Vulpes cana isolate B.F.Y3 cytochrome b (cytb) gene,
partial cds; mitochondrial</td>
<td align="left">KU378587</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Vulpes rueppellii</td>
<td align="left">Vulpes rueppellii isolate R.F.Y6 cytochrome b (cytb)
gene, partial cds; mitochondrial</td>
<td align="left">KU378373</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Vulpes lagopus</td>
<td align="left">Vulpes lagopus haplotype 5 cytochrome b (cytb) gene,
partial cds; mitochondrial</td>
<td align="left">KX093945</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Vulpes ferrilata</td>
<td align="left">Vulpes ferrilata haplotype 1 cytochrome b (cytb) gene,
partial cds; mitochondrial</td>
<td align="left">EU872065</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Vulpes macrotis</td>
<td align="left">Vulpes macrotis cytochrome b (cytb) gene, mitochondrial
gene encoding mitochondrial protein, partial cds</td>
<td align="left">AF028157</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Vulpes pallida</td>
<td align="left">Vulpes pallida haplotype PMa2 cytochrome b (cytb) gene,
partial cds; mitochondrial</td>
<td align="left">KJ597964</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">V.vulpes mitochondrial</td>
<td align="left">V.vulpes mitochondrial DNA for cytochrome b (complete
sequence)</td>
<td align="left">X94929</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Phoca largha</td>
<td align="left">Phoca largha PLCBRe4 mitochondrial cytb gene for
cytochrome b, partial cds</td>
<td align="left">LC466149</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Pagophilus groenlandicus</td>
<td align="left">Pagophilus groenlandicus cytochrome b gene, partial
cds; mitochondrial gene for mitochondrial product</td>
<td align="left">AF200491</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Phoca groenlandica</td>
<td align="left">Phoca groenlandica cytochrome b (cytb) gene, complete
cds; mitochondrial</td>
<td align="left">GU174609</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Phoca fasciata</td>
<td align="left">Phoca fasciata cytochrome b (cytb) gene, complete cds;
mitochondrial</td>
<td align="left">GU167294</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Phoca vitulina</td>
<td align="left">Phoca vitulina mitochondrial cytochrome b gene, partial
cds</td>
<td align="left">L19127</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">P.vitulina mitochondrial</td>
<td align="left">P.vitulina mitochondrial cytochrome b gene</td>
<td align="left">X82306</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">P.largha mitochondrial</td>
<td align="left">P.largha mitochondrial cytochrome b gene</td>
<td align="left">X82305</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">P.groenlandica mitochondrial</td>
<td align="left">P.groenlandica mitochondrial cytochrome b gene</td>
<td align="left">X82303</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">P.fasciata mitochondrial</td>
<td align="left">P.fasciata mitochondrial cytochrome b gene</td>
<td align="left">X82302</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Manis pentadactyla</td>
<td align="left">Manis pentadactyla isolate S105 cytochrome b (cytb)
gene, partial cds; mitochondrial</td>
<td align="left">MW197468</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Felis silvestris</td>
<td align="left">Felis silvestris silvestris isolate CAT155 NADH
dehydrogenase subunit 5 (ND5) and NADH dehydrogenase subunit 6 (ND6)
genes, partial cds; tRNA-Glu (trnE(ttc)) gene, partial sequence; and
cytochrome b (cytb) gene, partial cds; mitochondrial</td>
<td align="left">MN537981</td>
<td align="left">NADH dehydrogenase subunit 5</td>
</tr>
<tr class="odd">
<td align="left">Felis catus</td>
<td align="left">Felis catus isolate 4709-K NADH dehydrogenase subunit 5
(ND5) gene, partial cds; NADH dehydrogenase subunit 6 (ND6) gene,
complete cds; tRNA-Glu gene, complete sequence; and cytochrome b (CYTB)
gene, partial cds; mitochondrial</td>
<td align="left">MN313781</td>
<td align="left">NADH dehydrogenase subunit 5</td>
</tr>
<tr class="even">
<td align="left">Felis margarita</td>
<td align="left">Felis margarita haplotype AH NADH dehydrogenase subunit
5 (ND5) and cytochrome b (cytb) genes, partial cds; and tRNA-Thr gene
and D-loop, partial sequence; mitochondrial</td>
<td align="left">MK606132</td>
<td align="left">NADH dehydrogenase subunit 5</td>
</tr>
<tr class="odd">
<td align="left">Felis chaus</td>
<td align="left">Felis chaus isolate JCAIZ003 NADH dehydrogenase subunit
5 (ND5) gene, partial cds; mitochondrial</td>
<td align="left">GU561700</td>
<td align="left">NADH dehydrogenase subunit 5</td>
</tr>
<tr class="even">
<td align="left">Felis nigripes</td>
<td align="left">Felis nigripes NADH dehydrogenase subunit 5 gene,
partial cds; mitochondrial</td>
<td align="left">AF006400</td>
<td align="left">NADH dehydrogenase subunit 5</td>
</tr>
<tr class="odd">
<td align="left">Felis libyca</td>
<td align="left">Felis libyca NADH dehydrogenase subunit 5 gene, partial
cds; mitochondrial</td>
<td align="left">AF006396</td>
<td align="left">NADH dehydrogenase subunit 5</td>
</tr>
<tr class="even">
<td align="left">Vulpes lagopus</td>
<td align="left">Vulpes lagopus ATP synthase F0 subunit 6 (ATP6), ATP
synthase F0 subunit 8 (ATP8), cytochrome c oxidase subunit I (COX1),
cytochrome c oxidase subunit II (COX2), cytochrome c oxidase subunit III
(COX3), cytochrome b (CYTB), NADH dehydrogenase subunit 1 (ND1), NADH
dehydrogenase subunit 2 (ND2), NADH dehydrogenase subunit 3 (ND3), NADH
dehydrogenase subunit 4 (ND4), NADH dehydrogenase subunit 4L (ND4L), and
NADH dehydrogenase subunit 5 (ND5) genes, complete cds;
mitochondrial</td>
<td align="left">AH014073</td>
<td align="left">NADH dehydrogenase subunit 5</td>
</tr>
<tr class="odd">
<td align="left">Phoca groenlandica</td>
<td align="left">Phoca groenlandica NADH dehydrogenase subunit 5 (ND5)
gene, complete cds; mitochondrial gene for mitochondrial product</td>
<td align="left">AY377376</td>
<td align="left">NADH dehydrogenase subunit 5</td>
</tr>
<tr class="even">
<td align="left">Phoca fasciata</td>
<td align="left">Phoca fasciata NADH dehydrogenase subunit 5 (ND5) gene,
complete cds; mitochondrial</td>
<td align="left">GU167331</td>
<td align="left">NADH dehydrogenase subunit 5</td>
</tr>
<tr class="odd">
<td align="left">Felis silvestris</td>
<td align="left">Felis silvestris voucher 77_Fe_sil cytochrome c oxidase
subunit I (COX1) gene, partial sequence; nuclear copy of mitochondrial
gene</td>
<td align="left">KF297804</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="even">
<td align="left">Felis catus</td>
<td align="left">Felis catus voucher 191_Fe_cat cytochrome c oxidase
subunit I (COX1) gene, partial sequence; nuclear copy of mitochondrial
gene</td>
<td align="left">KF297802</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="odd">
<td align="left">Felis chaus</td>
<td align="left">Felis chaus cytochrome c oxidase subunit I (COI) gene,
partial cds; mitochondrial</td>
<td align="left">KJ634464</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="even">
<td align="left">Vulpes corsac</td>
<td align="left">Vulpes corsac isolate MGL-309 cytochrome c oxidase
subunit I (COX1) gene, partial cds; mitochondrial</td>
<td align="left">ON377297</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="odd">
<td align="left">Vulpes vulpes</td>
<td align="left">Vulpes vulpes isolate EXT423_2020 cytochrome c oxidase
subunit I (COX1) gene, partial cds; mitochondrial</td>
<td align="left">OM980695</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="even">
<td align="left">Vulpes lagopus</td>
<td align="left">Vulpes lagopus isolate VL1 cytochrome c oxidase subunit
I (COX1) gene, partial cds; mitochondrial</td>
<td align="left">MT731651</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="odd">
<td align="left">Alopex lagopus</td>
<td align="left">Alopex lagopus cytochrome c oxidase subunit I (COI)
gene, partial cds; mitochondrial</td>
<td align="left">AY894421</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="even">
<td align="left">Vulpes macrotis</td>
<td align="left">Vulpes macrotis cytochrome c oxidase subunit I (COI)
gene, mitochondrial gene encoding mitochondrial protein, partial
cds</td>
<td align="left">AF028205</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="odd">
<td align="left">Vulpes zerda</td>
<td align="left">Vulpes zerda cytochrome c oxidase subunit I (COI) gene,
mitochondrial gene encoding mitochondrial protein, partial cds</td>
<td align="left">AF028194</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="even">
<td align="left">Phoca vitulina</td>
<td align="left">Phoca vitulina vitulina isolate Atlantic harbor seal D
cytochrome c oxidase subunit I (COX1) gene, partial cds; tRNA-Ser and
tRNA-Asp genes, complete sequence; and cytochrome c oxidase subunit II
(COX2) gene, complete cds; mitochondrial</td>
<td align="left">GU733705</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="odd">
<td align="left">Manis pentadactyla</td>
<td align="left">Manis pentadactyla pentadactyla isolate ASIZFZ002032
cytochrome c oxidase subunit I (COX1) gene, partial cds;
mitochondrial</td>
<td align="left">OM220069</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="even">
<td align="left">Felis catus</td>
<td align="left">Felis catus 16S ribosomal RNA gene, partial sequence;
mitochondrial</td>
<td align="left">AY011183</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="odd">
<td align="left">Felis silvestris</td>
<td align="left">Felis silvestris isolate A1 16S ribosomal RNA gene,
partial sequence; mitochondrial</td>
<td align="left">MF444948</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="even">
<td align="left">Felis libyca</td>
<td align="left">Felis libyca 16S ribosomal RNA gene, mitochondrial gene
for mitochondrial RNA, partial sequence</td>
<td align="left">AF006395</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="odd">
<td align="left">Felis nigripes</td>
<td align="left">Felis nigripes 16S ribosomal RNA gene, mitochondrial
gene for mitochondrial RNA, partial sequence</td>
<td align="left">AF006399</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="even">
<td align="left">Felis margarita</td>
<td align="left">Felis margarita 16S ribosomal RNA gene, mitochondrial
gene for mitochondrial RNA, partial sequence</td>
<td align="left">AF006397</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="odd">
<td align="left">Felis chaus</td>
<td align="left">Felis chaus 16S ribosomal RNA gene, mitochondrial gene
for mitochondrial RNA, partial sequence</td>
<td align="left">AF006393</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="even">
<td align="left">Vulpes vulpes</td>
<td align="left">Vulpes vulpes voucher UAM:Mamm:24948 16S ribosomal RNA
gene, partial sequence; mitochondrial</td>
<td align="left">DQ334815</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="odd">
<td align="left">Vulpes lagopus</td>
<td align="left">Vulpes lagopus voucher UAM:Mamm:22740 16S ribosomal RNA
gene, partial sequence; mitochondrial</td>
<td align="left">DQ334814</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="even">
<td align="left">Vulpes ferrilata</td>
<td align="left">Vulpes ferrilata voucher QTPNM 12112003 16S ribosomal
RNA gene, partial sequence; mitochondrial</td>
<td align="left">KC538826</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="odd">
<td align="left">Manis pentadactyla</td>
<td align="left">Manis pentadactyla 16S ribosomal RNA gene, partial
sequence; mitochondrial</td>
<td align="left">AY011188</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="even">
<td align="left">Felis catus</td>
<td align="left">Felis catus voucher N22b 12S ribosomal RNA gene,
partial sequence; mitochondrial</td>
<td align="left">KX786344</td>
<td align="left">12S ribosomal RNA</td>
</tr>
<tr class="odd">
<td align="left">Felis chaus</td>
<td align="left">Felis chaus isolate G 12S ribosomal RNA gene, partial
sequence; mitochondrial</td>
<td align="left">KU963205</td>
<td align="left">12S ribosomal RNA</td>
</tr>
<tr class="even">
<td align="left">Felis silvestris</td>
<td align="left">Felis silvestris 12S ribosomal RNA gene, partial
sequence; mitochondrial</td>
<td align="left">KX002032</td>
<td align="left">12S ribosomal RNA</td>
</tr>
<tr class="odd">
<td align="left">Felis bieti</td>
<td align="left">Felis bieti 12S ribosomal RNA gene, partial sequence;
mitochondrial</td>
<td align="left">AY773084</td>
<td align="left">12S ribosomal RNA</td>
</tr>
<tr class="even">
<td align="left">Vulpes vulpes</td>
<td align="left">Vulpes vulpes Vv1 mitochondrial gene for 12S ribosomal
RNA, partial sequence</td>
<td align="left">LC424764</td>
<td align="left">12S ribosomal RNA</td>
</tr>
<tr class="odd">
<td align="left">Vulpes lagopus</td>
<td align="left">Vulpes lagopus isolate FRT12 12S ribosomal RNA gene,
partial sequence; mitochondrial</td>
<td align="left">KM224240</td>
<td align="left">12S ribosomal RNA</td>
</tr>
<tr class="even">
<td align="left">Phoca fasciata</td>
<td align="left">Phoca fasciata isolate 5888 12S ribosomal RNA gene,
partial sequence; mitochondrial</td>
<td align="left">GU174595</td>
<td align="left">12S ribosomal RNA</td>
</tr>
<tr class="odd">
<td align="left">Phoca largha</td>
<td align="left">Phoca largha isolate 06spotted03 12S ribosomal RNA
gene, partial sequence; mitochondrial</td>
<td align="left">GU174591</td>
<td align="left">12S ribosomal RNA</td>
</tr>
<tr class="even">
<td align="left">Manis pentadactyla</td>
<td align="left">Manis pentadactyla 12S ribosomal RNA gene, partial
sequence; and tRNA-Val gene, complete sequence; mitochondrial</td>
<td align="left">AY012154</td>
<td align="left">12S ribosomal RNA</td>
</tr>
</tbody>
</table>
<p>Feel free to review this dataset, make changes, add new species,
samples, etc. The integrity of this dataset is critical for the next
steps so please take your time and review it carefully. For instance,
let‚Äôs just make some minor changes to our dataset:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">acc.table</span><span class="op">$</span><span class="va">Species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"P."</span>, <span class="st">"Phoca "</span>, <span class="va">acc.table</span><span class="op">$</span><span class="va">Species</span>, fixed <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">acc.table</span><span class="op">$</span><span class="va">Species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"F."</span>, <span class="st">"Felis "</span>, <span class="va">acc.table</span><span class="op">$</span><span class="va">Species</span>, fixed <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">acc.table</span><span class="op">$</span><span class="va">Species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"V."</span>, <span class="st">"Vulpes "</span>, <span class="va">acc.table</span><span class="op">$</span><span class="va">Species</span>, fixed <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">acc.table</span><span class="op">$</span><span class="va">Species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">sub</a></span><span class="op">(</span><span class="st">"mitochondrial"</span>, <span class="st">""</span>, <span class="va">acc.table</span><span class="op">$</span><span class="va">Species</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/row.names.html" class="external-link">row.names</a></span><span class="op">(</span><span class="va">acc.table</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></code></pre></div>
<p>Let‚Äôs check how the new table looks now‚Ä¶</p>
<table class="table">
<caption>Updated Table 2 with some minor corrections on species names.
Other updates to this table include adding more sequences, using
specific sequences (instead of the ones suggested by
<code>phruta</code>).</caption>
<colgroup>
<col width="4%">
<col width="88%">
<col width="1%">
<col width="5%">
</colgroup>
<thead><tr class="header">
<th align="left">Species</th>
<th align="left">Ti</th>
<th align="left">Acc</th>
<th align="left">gene</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Felis catus</td>
<td align="left">Felis catus MKRaS006 mitochondrial gene for cytochrome
b, partial cds</td>
<td align="left">LC649704</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Felis silvestris</td>
<td align="left">Felis silvestris lybica isolate CAT157 NADH
dehydrogenase subunit 5 (ND5) gene, partial cds; NADH dehydrogenase
subunit 6 (ND6) gene, complete cds; tRNA-Glu (trnE(ttc)) gene, complete
sequence; and cytochrome b (cytb) gene, partial cds; mitochondrial</td>
<td align="left">MN537982</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Felis chaus</td>
<td align="left">Felis chaus isolate Jungle Cat 5 cytochrome b (cytb)
gene, partial cds; mitochondrial</td>
<td align="left">MN370575</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Felis environmental</td>
<td align="left">Felis environmental sample isolate Kw-170 cytochrome b
(cytb) gene, partial cds; mitochondrial</td>
<td align="left">MK510873</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Felis margarita</td>
<td align="left">Felis margarita haplotype AH NADH dehydrogenase subunit
5 (ND5) and cytochrome b (cytb) genes, partial cds; and tRNA-Thr gene
and D-loop, partial sequence; mitochondrial</td>
<td align="left">MK606132</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Felis bieti</td>
<td align="left">Felis bieti cytochrome b gene, partial cds;
mitochondrial</td>
<td align="left">AY773081</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Felis domesticus</td>
<td align="left">F.domesticus mitochondrial cytochrome b gene</td>
<td align="left">X82296</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Vulpes corsac</td>
<td align="left">Vulpes corsac isolate SH21 cytochrome b (CYTB) gene,
complete cds; mitochondrial</td>
<td align="left">MT795179</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Vulpes vulpes</td>
<td align="left">Vulpes vulpes isolate Kharkiv10 cytochrome b (cytb)
gene, complete cds; mitochondrial</td>
<td align="left">EF689062</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Vulpes zerda</td>
<td align="left">Vulpes zerda isolate X161349 cytochrome b (Cytb) gene,
partial cds; mitochondrial</td>
<td align="left">MH854561</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Vulpes cana</td>
<td align="left">Vulpes cana isolate B.F.Y3 cytochrome b (cytb) gene,
partial cds; mitochondrial</td>
<td align="left">KU378587</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Vulpes rueppellii</td>
<td align="left">Vulpes rueppellii isolate R.F.Y6 cytochrome b (cytb)
gene, partial cds; mitochondrial</td>
<td align="left">KU378373</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Vulpes lagopus</td>
<td align="left">Vulpes lagopus haplotype 5 cytochrome b (cytb) gene,
partial cds; mitochondrial</td>
<td align="left">KX093945</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Vulpes ferrilata</td>
<td align="left">Vulpes ferrilata haplotype 1 cytochrome b (cytb) gene,
partial cds; mitochondrial</td>
<td align="left">EU872065</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Vulpes macrotis</td>
<td align="left">Vulpes macrotis cytochrome b (cytb) gene, mitochondrial
gene encoding mitochondrial protein, partial cds</td>
<td align="left">AF028157</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Vulpes pallida</td>
<td align="left">Vulpes pallida haplotype PMa2 cytochrome b (cytb) gene,
partial cds; mitochondrial</td>
<td align="left">KJ597964</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Vulpes vulpes</td>
<td align="left">V.vulpes mitochondrial DNA for cytochrome b (complete
sequence)</td>
<td align="left">X94929</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Phoca largha</td>
<td align="left">Phoca largha PLCBRe4 mitochondrial cytb gene for
cytochrome b, partial cds</td>
<td align="left">LC466149</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Pagophilus groenlandicus</td>
<td align="left">Pagophilus groenlandicus cytochrome b gene, partial
cds; mitochondrial gene for mitochondrial product</td>
<td align="left">AF200491</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Phoca groenlandica</td>
<td align="left">Phoca groenlandica cytochrome b (cytb) gene, complete
cds; mitochondrial</td>
<td align="left">GU174609</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Phoca fasciata</td>
<td align="left">Phoca fasciata cytochrome b (cytb) gene, complete cds;
mitochondrial</td>
<td align="left">GU167294</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Phoca vitulina</td>
<td align="left">Phoca vitulina mitochondrial cytochrome b gene, partial
cds</td>
<td align="left">L19127</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Phoca vitulina</td>
<td align="left">P.vitulina mitochondrial cytochrome b gene</td>
<td align="left">X82306</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Phoca largha</td>
<td align="left">P.largha mitochondrial cytochrome b gene</td>
<td align="left">X82305</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Phoca groenlandica</td>
<td align="left">P.groenlandica mitochondrial cytochrome b gene</td>
<td align="left">X82303</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Phoca fasciata</td>
<td align="left">P.fasciata mitochondrial cytochrome b gene</td>
<td align="left">X82302</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="odd">
<td align="left">Manis pentadactyla</td>
<td align="left">Manis pentadactyla isolate S105 cytochrome b (cytb)
gene, partial cds; mitochondrial</td>
<td align="left">MW197468</td>
<td align="left">cytochrome b</td>
</tr>
<tr class="even">
<td align="left">Felis silvestris</td>
<td align="left">Felis silvestris silvestris isolate CAT155 NADH
dehydrogenase subunit 5 (ND5) and NADH dehydrogenase subunit 6 (ND6)
genes, partial cds; tRNA-Glu (trnE(ttc)) gene, partial sequence; and
cytochrome b (cytb) gene, partial cds; mitochondrial</td>
<td align="left">MN537981</td>
<td align="left">NADH dehydrogenase subunit 5</td>
</tr>
<tr class="odd">
<td align="left">Felis catus</td>
<td align="left">Felis catus isolate 4709-K NADH dehydrogenase subunit 5
(ND5) gene, partial cds; NADH dehydrogenase subunit 6 (ND6) gene,
complete cds; tRNA-Glu gene, complete sequence; and cytochrome b (CYTB)
gene, partial cds; mitochondrial</td>
<td align="left">MN313781</td>
<td align="left">NADH dehydrogenase subunit 5</td>
</tr>
<tr class="even">
<td align="left">Felis margarita</td>
<td align="left">Felis margarita haplotype AH NADH dehydrogenase subunit
5 (ND5) and cytochrome b (cytb) genes, partial cds; and tRNA-Thr gene
and D-loop, partial sequence; mitochondrial</td>
<td align="left">MK606132</td>
<td align="left">NADH dehydrogenase subunit 5</td>
</tr>
<tr class="odd">
<td align="left">Felis chaus</td>
<td align="left">Felis chaus isolate JCAIZ003 NADH dehydrogenase subunit
5 (ND5) gene, partial cds; mitochondrial</td>
<td align="left">GU561700</td>
<td align="left">NADH dehydrogenase subunit 5</td>
</tr>
<tr class="even">
<td align="left">Felis nigripes</td>
<td align="left">Felis nigripes NADH dehydrogenase subunit 5 gene,
partial cds; mitochondrial</td>
<td align="left">AF006400</td>
<td align="left">NADH dehydrogenase subunit 5</td>
</tr>
<tr class="odd">
<td align="left">Felis libyca</td>
<td align="left">Felis libyca NADH dehydrogenase subunit 5 gene, partial
cds; mitochondrial</td>
<td align="left">AF006396</td>
<td align="left">NADH dehydrogenase subunit 5</td>
</tr>
<tr class="even">
<td align="left">Vulpes lagopus</td>
<td align="left">Vulpes lagopus ATP synthase F0 subunit 6 (ATP6), ATP
synthase F0 subunit 8 (ATP8), cytochrome c oxidase subunit I (COX1),
cytochrome c oxidase subunit II (COX2), cytochrome c oxidase subunit III
(COX3), cytochrome b (CYTB), NADH dehydrogenase subunit 1 (ND1), NADH
dehydrogenase subunit 2 (ND2), NADH dehydrogenase subunit 3 (ND3), NADH
dehydrogenase subunit 4 (ND4), NADH dehydrogenase subunit 4L (ND4L), and
NADH dehydrogenase subunit 5 (ND5) genes, complete cds;
mitochondrial</td>
<td align="left">AH014073</td>
<td align="left">NADH dehydrogenase subunit 5</td>
</tr>
<tr class="odd">
<td align="left">Phoca groenlandica</td>
<td align="left">Phoca groenlandica NADH dehydrogenase subunit 5 (ND5)
gene, complete cds; mitochondrial gene for mitochondrial product</td>
<td align="left">AY377376</td>
<td align="left">NADH dehydrogenase subunit 5</td>
</tr>
<tr class="even">
<td align="left">Phoca fasciata</td>
<td align="left">Phoca fasciata NADH dehydrogenase subunit 5 (ND5) gene,
complete cds; mitochondrial</td>
<td align="left">GU167331</td>
<td align="left">NADH dehydrogenase subunit 5</td>
</tr>
<tr class="odd">
<td align="left">Felis silvestris</td>
<td align="left">Felis silvestris voucher 77_Fe_sil cytochrome c oxidase
subunit I (COX1) gene, partial sequence; nuclear copy of mitochondrial
gene</td>
<td align="left">KF297804</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="even">
<td align="left">Felis catus</td>
<td align="left">Felis catus voucher 191_Fe_cat cytochrome c oxidase
subunit I (COX1) gene, partial sequence; nuclear copy of mitochondrial
gene</td>
<td align="left">KF297802</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="odd">
<td align="left">Felis chaus</td>
<td align="left">Felis chaus cytochrome c oxidase subunit I (COI) gene,
partial cds; mitochondrial</td>
<td align="left">KJ634464</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="even">
<td align="left">Vulpes corsac</td>
<td align="left">Vulpes corsac isolate MGL-309 cytochrome c oxidase
subunit I (COX1) gene, partial cds; mitochondrial</td>
<td align="left">ON377297</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="odd">
<td align="left">Vulpes vulpes</td>
<td align="left">Vulpes vulpes isolate EXT423_2020 cytochrome c oxidase
subunit I (COX1) gene, partial cds; mitochondrial</td>
<td align="left">OM980695</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="even">
<td align="left">Vulpes lagopus</td>
<td align="left">Vulpes lagopus isolate VL1 cytochrome c oxidase subunit
I (COX1) gene, partial cds; mitochondrial</td>
<td align="left">MT731651</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="odd">
<td align="left">Alopex lagopus</td>
<td align="left">Alopex lagopus cytochrome c oxidase subunit I (COI)
gene, partial cds; mitochondrial</td>
<td align="left">AY894421</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="even">
<td align="left">Vulpes macrotis</td>
<td align="left">Vulpes macrotis cytochrome c oxidase subunit I (COI)
gene, mitochondrial gene encoding mitochondrial protein, partial
cds</td>
<td align="left">AF028205</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="odd">
<td align="left">Vulpes zerda</td>
<td align="left">Vulpes zerda cytochrome c oxidase subunit I (COI) gene,
mitochondrial gene encoding mitochondrial protein, partial cds</td>
<td align="left">AF028194</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="even">
<td align="left">Phoca vitulina</td>
<td align="left">Phoca vitulina vitulina isolate Atlantic harbor seal D
cytochrome c oxidase subunit I (COX1) gene, partial cds; tRNA-Ser and
tRNA-Asp genes, complete sequence; and cytochrome c oxidase subunit II
(COX2) gene, complete cds; mitochondrial</td>
<td align="left">GU733705</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="odd">
<td align="left">Manis pentadactyla</td>
<td align="left">Manis pentadactyla pentadactyla isolate ASIZFZ002032
cytochrome c oxidase subunit I (COX1) gene, partial cds;
mitochondrial</td>
<td align="left">OM220069</td>
<td align="left">cytochrome c oxidase subunit I</td>
</tr>
<tr class="even">
<td align="left">Felis catus</td>
<td align="left">Felis catus 16S ribosomal RNA gene, partial sequence;
mitochondrial</td>
<td align="left">AY011183</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="odd">
<td align="left">Felis silvestris</td>
<td align="left">Felis silvestris isolate A1 16S ribosomal RNA gene,
partial sequence; mitochondrial</td>
<td align="left">MF444948</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="even">
<td align="left">Felis libyca</td>
<td align="left">Felis libyca 16S ribosomal RNA gene, mitochondrial gene
for mitochondrial RNA, partial sequence</td>
<td align="left">AF006395</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="odd">
<td align="left">Felis nigripes</td>
<td align="left">Felis nigripes 16S ribosomal RNA gene, mitochondrial
gene for mitochondrial RNA, partial sequence</td>
<td align="left">AF006399</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="even">
<td align="left">Felis margarita</td>
<td align="left">Felis margarita 16S ribosomal RNA gene, mitochondrial
gene for mitochondrial RNA, partial sequence</td>
<td align="left">AF006397</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="odd">
<td align="left">Felis chaus</td>
<td align="left">Felis chaus 16S ribosomal RNA gene, mitochondrial gene
for mitochondrial RNA, partial sequence</td>
<td align="left">AF006393</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="even">
<td align="left">Vulpes vulpes</td>
<td align="left">Vulpes vulpes voucher UAM:Mamm:24948 16S ribosomal RNA
gene, partial sequence; mitochondrial</td>
<td align="left">DQ334815</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="odd">
<td align="left">Vulpes lagopus</td>
<td align="left">Vulpes lagopus voucher UAM:Mamm:22740 16S ribosomal RNA
gene, partial sequence; mitochondrial</td>
<td align="left">DQ334814</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="even">
<td align="left">Vulpes ferrilata</td>
<td align="left">Vulpes ferrilata voucher QTPNM 12112003 16S ribosomal
RNA gene, partial sequence; mitochondrial</td>
<td align="left">KC538826</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="odd">
<td align="left">Manis pentadactyla</td>
<td align="left">Manis pentadactyla 16S ribosomal RNA gene, partial
sequence; mitochondrial</td>
<td align="left">AY011188</td>
<td align="left">16S ribosomal RNA</td>
</tr>
<tr class="even">
<td align="left">Felis catus</td>
<td align="left">Felis catus voucher N22b 12S ribosomal RNA gene,
partial sequence; mitochondrial</td>
<td align="left">KX786344</td>
<td align="left">12S ribosomal RNA</td>
</tr>
<tr class="odd">
<td align="left">Felis chaus</td>
<td align="left">Felis chaus isolate G 12S ribosomal RNA gene, partial
sequence; mitochondrial</td>
<td align="left">KU963205</td>
<td align="left">12S ribosomal RNA</td>
</tr>
<tr class="even">
<td align="left">Felis silvestris</td>
<td align="left">Felis silvestris 12S ribosomal RNA gene, partial
sequence; mitochondrial</td>
<td align="left">KX002032</td>
<td align="left">12S ribosomal RNA</td>
</tr>
<tr class="odd">
<td align="left">Felis bieti</td>
<td align="left">Felis bieti 12S ribosomal RNA gene, partial sequence;
mitochondrial</td>
<td align="left">AY773084</td>
<td align="left">12S ribosomal RNA</td>
</tr>
<tr class="even">
<td align="left">Vulpes vulpes</td>
<td align="left">Vulpes vulpes Vv1 mitochondrial gene for 12S ribosomal
RNA, partial sequence</td>
<td align="left">LC424764</td>
<td align="left">12S ribosomal RNA</td>
</tr>
<tr class="odd">
<td align="left">Vulpes lagopus</td>
<td align="left">Vulpes lagopus isolate FRT12 12S ribosomal RNA gene,
partial sequence; mitochondrial</td>
<td align="left">KM224240</td>
<td align="left">12S ribosomal RNA</td>
</tr>
<tr class="even">
<td align="left">Phoca fasciata</td>
<td align="left">Phoca fasciata isolate 5888 12S ribosomal RNA gene,
partial sequence; mitochondrial</td>
<td align="left">GU174595</td>
<td align="left">12S ribosomal RNA</td>
</tr>
<tr class="odd">
<td align="left">Phoca largha</td>
<td align="left">Phoca largha isolate 06spotted03 12S ribosomal RNA
gene, partial sequence; mitochondrial</td>
<td align="left">GU174591</td>
<td align="left">12S ribosomal RNA</td>
</tr>
<tr class="even">
<td align="left">Manis pentadactyla</td>
<td align="left">Manis pentadactyla 12S ribosomal RNA gene, partial
sequence; and tRNA-Val gene, complete sequence; mitochondrial</td>
<td align="left">AY012154</td>
<td align="left">12S ribosomal RNA</td>
</tr>
</tbody>
</table>
<p>Now, since we‚Äôre going to retrieve sequences from genbank using an
existing preliminary accession numbers table, we will use the
<code><a href="../reference/sq.retrieve.indirect.html">sq.retrieve.indirect()</a></code> function in <code>phruta</code>. I‚Äôm
going to spend some time in here to explain the differences between the
two versions of <code>sq.retrieve.*</code> in <code>phruta</code>. The
one that we‚Äôre using in this tutorial,
<code><a href="../reference/sq.retrieve.indirect.html">sq.retrieve.indirect()</a></code>, retrieves sequences ‚Äúindirectly‚Äù
because it follows the initial step of generating a table summarizing
accession numbers (see the <code><a href="../reference/acc.table.retrieve.html">acc.table.retrieve()</a></code> function
above). I present the information in this vignette using
<code><a href="../reference/sq.retrieve.indirect.html">sq.retrieve.indirect()</a></code> instead of
<code><a href="../reference/sq.retrieve.direct.html">sq.retrieve.direct()</a></code> because the first function is way more
flexible and allows for correcting issues <em>prior</em> to download any
sequence. For instance, you can add new sequences, species, populations
to the resulting data.frame from <code><a href="../reference/acc.table.retrieve.html">acc.table.retrieve()</a></code>.
Additionally, you could even manually assemble your own dataset of
accession numbers to be retrieved using
<code><a href="../reference/sq.retrieve.indirect.html">sq.retrieve.indirect()</a></code>. Instead,
<code><a href="../reference/sq.retrieve.direct.html">sq.retrieve.direct()</a></code> does its best to directly
(i.e.¬†without potential input from the user) retrieve sequences for a
target set of taxa and set of gene regions. In short, you should be able
to catch errors using <code><a href="../reference/sq.retrieve.indirect.html">sq.retrieve.indirect()</a></code> but mistakes
will be harder to spot and fix if you‚Äôre using
<code><a href="../reference/sq.retrieve.direct.html">sq.retrieve.direct()</a></code>. Note that the functionality of
<code><a href="../reference/sq.retrieve.direct.html">sq.retrieve.direct()</a></code> is outlined in the ‚ÄúUsing
<code>phruta</code> with defined target genes‚Äù vignette.</p>
<p>We still need to retrieve all the sequences from the accessions table
that was generated avobe using <code>acc.table</code>. The
<code><a href="../reference/sq.retrieve.indirect.html">sq.retrieve.indirect()</a></code> function will write all the
resulting <code>fasta</code> files into a newly created folder
<code>0.Sequences</code> located in our working directory (please check
the <code>download.sqs = TRUE</code> argument).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/sq.retrieve.indirect.html">sq.retrieve.indirect</a></span><span class="op">(</span><span class="va">acc.table</span>, download.sqs <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>Next, we‚Äôre going to make sure that we include only sequences that
are reliable and from species that we are actually interested in
analyzing. For this, we will be using the <code><a href="../reference/sq.curate.html">sq.curate()</a></code>
function. We need to provide a list of taxonomic names to filter out
incorrect sequences (<code>filterTaxonomicCriteria</code> argument). For
simplicity, our criteria can be the genera that we‚Äôre interested in
analyzing. Note that the outgroup‚Äôs name should also be included in the
list. If the taxonomic information for a sequence retrieved from genbank
does not match with any of these strings, this species will be dropped.
You will have to specify whether sampling is for animals or plants
(<code>kingdom</code> argument). Finally, you might have already noticed
that the same gene regions can have different names. For instance,
sometimes our searches retrieve both ‚Äúcytochrome oxidase subunit 1‚Äù and
‚Äúcytochrome c oxidase subunit I‚Äù as widely sampled genes for the target
species. In that case, we can combine the sequences in these two files
into a single file name <code>COI</code>. To merge gene files, you will
have to provide a named list to the <code>mergeGeneFiles</code> argument
of the <code>sq.curate</code> function. This named list
(<code>tb.merged</code> below) will have a length that corresponds to
the number of final files that should be constructed.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tb.merged</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">'COI'</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cytochrome oxidase subunit 1"</span>, <span class="st">"cytochrome c oxidase subunit I"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="../reference/sq.curate.html">sq.curate</a></span><span class="op">(</span>filterTaxonomicCriteria <span class="op">=</span> <span class="st">'Felis|Vulpes|Phoca|Manis'</span>,
          mergeGeneFiles <span class="op">=</span> <span class="va">tb.merged</span>,
          kingdom <span class="op">=</span> <span class="st">'animals'</span>, 
          folder <span class="op">=</span> <span class="st">'0.Sequences'</span>,
          removeOutliers <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>Running the line of code above will create the a folder
<code>1.CuratedSequences</code> containing (1) the curated sequences
with original names, (2) the curated sequences with species-level names
(<code>renamed_*</code> prefix), (3) a table of accession numbers
(<code>0.AccessionTable.csv</code>), and (4) a summary of the taxonomic
information for all the species sampled in the files
(<code>1.Taxonomy.csv</code>). We‚Äôll use the <code>renamed_*</code> and
<code>1.Taxonomy.csv</code> files in the next steps. Let‚Äôs take a look
at the sampling per gene region in the <code>0.AccessionTable.csv</code>
table.</p>
<table class="table">
<caption>4. Preliminary accession number table</caption>
<colgroup>
<col width="23%">
<col width="7%">
<col width="21%">
<col width="31%">
<col width="16%">
</colgroup>
<thead><tr class="header">
<th align="left">OriginalNames</th>
<th align="left">AccN</th>
<th align="left">Species</th>
<th align="left">file</th>
<th align="left">OldSpecies</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">KX786344 Felis catus</td>
<td align="left">KX786344</td>
<td align="left">Felis_catus</td>
<td align="left">12S ribosomal RNA.fasta</td>
<td align="left">Felis_catus</td>
</tr>
<tr class="even">
<td align="left">KU963205 Felis chaus</td>
<td align="left">KU963205</td>
<td align="left">Felis_chaus</td>
<td align="left">12S ribosomal RNA.fasta</td>
<td align="left">Felis_chaus</td>
</tr>
<tr class="odd">
<td align="left">KX002032 Felis silvestris</td>
<td align="left">KX002032</td>
<td align="left">Felis_silvestris</td>
<td align="left">12S ribosomal RNA.fasta</td>
<td align="left">Felis_silvestris</td>
</tr>
<tr class="even">
<td align="left">AY773084 Felis bieti</td>
<td align="left">AY773084</td>
<td align="left">Felis_bieti</td>
<td align="left">12S ribosomal RNA.fasta</td>
<td align="left">Felis_bieti</td>
</tr>
<tr class="odd">
<td align="left">LC424764 Vulpes vulpes</td>
<td align="left">LC424764</td>
<td align="left">Vulpes_vulpes</td>
<td align="left">12S ribosomal RNA.fasta</td>
<td align="left">Vulpes_vulpes</td>
</tr>
<tr class="even">
<td align="left">KM224240 Vulpes lagopus</td>
<td align="left">KM224240</td>
<td align="left">Vulpes_lagopus</td>
<td align="left">12S ribosomal RNA.fasta</td>
<td align="left">Vulpes_lagopus</td>
</tr>
<tr class="odd">
<td align="left">GU174595 Phoca fasciata</td>
<td align="left">GU174595</td>
<td align="left">Histriophoca_fasciata</td>
<td align="left">12S ribosomal RNA.fasta</td>
<td align="left">Phoca_fasciata</td>
</tr>
<tr class="even">
<td align="left">GU174591 Phoca largha</td>
<td align="left">GU174591</td>
<td align="left">Phoca_largha</td>
<td align="left">12S ribosomal RNA.fasta</td>
<td align="left">Phoca_largha</td>
</tr>
<tr class="odd">
<td align="left">AY012154 Manis pentadactyla</td>
<td align="left">AY012154</td>
<td align="left">Manis_pentadactyla</td>
<td align="left">12S ribosomal RNA.fasta</td>
<td align="left">Manis_pentadactyla</td>
</tr>
<tr class="even">
<td align="left">AY011183 Felis catus</td>
<td align="left">AY011183</td>
<td align="left">Felis_catus</td>
<td align="left">16S ribosomal RNA.fasta</td>
<td align="left">Felis_catus</td>
</tr>
<tr class="odd">
<td align="left">MF444948 Felis silvestris</td>
<td align="left">MF444948</td>
<td align="left">Felis_silvestris</td>
<td align="left">16S ribosomal RNA.fasta</td>
<td align="left">Felis_silvestris</td>
</tr>
<tr class="even">
<td align="left">AF006399 Felis nigripes</td>
<td align="left">AF006399</td>
<td align="left">Felis_nigripes</td>
<td align="left">16S ribosomal RNA.fasta</td>
<td align="left">Felis_nigripes</td>
</tr>
<tr class="odd">
<td align="left">AF006397 Felis margarita</td>
<td align="left">AF006397</td>
<td align="left">Felis_margarita</td>
<td align="left">16S ribosomal RNA.fasta</td>
<td align="left">Felis_margarita</td>
</tr>
<tr class="even">
<td align="left">AF006393 Felis chaus</td>
<td align="left">AF006393</td>
<td align="left">Felis_chaus</td>
<td align="left">16S ribosomal RNA.fasta</td>
<td align="left">Felis_chaus</td>
</tr>
<tr class="odd">
<td align="left">DQ334815 Vulpes vulpes</td>
<td align="left">DQ334815</td>
<td align="left">Vulpes_vulpes</td>
<td align="left">16S ribosomal RNA.fasta</td>
<td align="left">Vulpes_vulpes</td>
</tr>
<tr class="even">
<td align="left">DQ334814 Vulpes lagopus</td>
<td align="left">DQ334814</td>
<td align="left">Vulpes_lagopus</td>
<td align="left">16S ribosomal RNA.fasta</td>
<td align="left">Vulpes_lagopus</td>
</tr>
<tr class="odd">
<td align="left">KC538826 Vulpes ferrilata</td>
<td align="left">KC538826</td>
<td align="left">Vulpes_ferrilata</td>
<td align="left">16S ribosomal RNA.fasta</td>
<td align="left">Vulpes_ferrilata</td>
</tr>
<tr class="even">
<td align="left">AY011188 Manis pentadactyla</td>
<td align="left">AY011188</td>
<td align="left">Manis_pentadactyla</td>
<td align="left">16S ribosomal RNA.fasta</td>
<td align="left">Manis_pentadactyla</td>
</tr>
<tr class="odd">
<td align="left">LC649704 Felis catus</td>
<td align="left">LC649704</td>
<td align="left">Felis_catus</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Felis_catus</td>
</tr>
<tr class="even">
<td align="left">MN537982 Felis silvestris</td>
<td align="left">MN537982</td>
<td align="left">Felis_silvestris</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Felis_silvestris</td>
</tr>
<tr class="odd">
<td align="left">MN370575 Felis chaus</td>
<td align="left">MN370575</td>
<td align="left">Felis_chaus</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Felis_chaus</td>
</tr>
<tr class="even">
<td align="left">MK606132 Felis margarita</td>
<td align="left">MK606132</td>
<td align="left">Felis_margarita</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Felis_margarita</td>
</tr>
<tr class="odd">
<td align="left">AY773081 Felis bieti</td>
<td align="left">AY773081</td>
<td align="left">Felis_bieti</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Felis_bieti</td>
</tr>
<tr class="even">
<td align="left">MT795179 Vulpes corsac</td>
<td align="left">MT795179</td>
<td align="left">Vulpes_corsac</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Vulpes_corsac</td>
</tr>
<tr class="odd">
<td align="left">EF689062 Vulpes vulpes</td>
<td align="left">EF689062</td>
<td align="left">Vulpes_vulpes</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Vulpes_vulpes</td>
</tr>
<tr class="even">
<td align="left">MH854561 Vulpes zerda</td>
<td align="left">MH854561</td>
<td align="left">Vulpes_zerda</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Vulpes_zerda</td>
</tr>
<tr class="odd">
<td align="left">KU378587 Vulpes cana</td>
<td align="left">KU378587</td>
<td align="left">Vulpes_cana</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Vulpes_cana</td>
</tr>
<tr class="even">
<td align="left">KU378373 Vulpes rueppellii</td>
<td align="left">KU378373</td>
<td align="left">Vulpes_rueppellii</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Vulpes_rueppellii</td>
</tr>
<tr class="odd">
<td align="left">KX093945 Vulpes lagopus</td>
<td align="left">KX093945</td>
<td align="left">Vulpes_lagopus</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Vulpes_lagopus</td>
</tr>
<tr class="even">
<td align="left">EU872065 Vulpes ferrilata</td>
<td align="left">EU872065</td>
<td align="left">Vulpes_ferrilata</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Vulpes_ferrilata</td>
</tr>
<tr class="odd">
<td align="left">AF028157 Vulpes macrotis</td>
<td align="left">AF028157</td>
<td align="left">Vulpes_macrotis</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Vulpes_macrotis</td>
</tr>
<tr class="even">
<td align="left">KJ597964 Vulpes pallida</td>
<td align="left">KJ597964</td>
<td align="left">Vulpes_pallida</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Vulpes_pallida</td>
</tr>
<tr class="odd">
<td align="left">LC466149 Phoca largha</td>
<td align="left">LC466149</td>
<td align="left">Phoca_largha</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Phoca_largha</td>
</tr>
<tr class="even">
<td align="left">GU174609 Phoca groenlandica</td>
<td align="left">GU174609</td>
<td align="left">Pagophilus_groenlandicus</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Phoca_groenlandica</td>
</tr>
<tr class="odd">
<td align="left">GU167294 Phoca fasciata</td>
<td align="left">GU167294</td>
<td align="left">Histriophoca_fasciata</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Phoca_fasciata</td>
</tr>
<tr class="even">
<td align="left">L19127 Phoca vitulina</td>
<td align="left">L19127</td>
<td align="left">Phoca_vitulina</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Phoca_vitulina</td>
</tr>
<tr class="odd">
<td align="left">MW197468 Manis pentadactyla</td>
<td align="left">MW197468</td>
<td align="left">Manis_pentadactyla</td>
<td align="left">cytochrome b.fasta</td>
<td align="left">Manis_pentadactyla</td>
</tr>
<tr class="even">
<td align="left">KF297804 Felis silvestris</td>
<td align="left">KF297804</td>
<td align="left">Felis_silvestris</td>
<td align="left">cytochrome c oxidase subunit I.fasta</td>
<td align="left">Felis_silvestris</td>
</tr>
<tr class="odd">
<td align="left">KF297802 Felis catus</td>
<td align="left">KF297802</td>
<td align="left">Felis_catus</td>
<td align="left">cytochrome c oxidase subunit I.fasta</td>
<td align="left">Felis_catus</td>
</tr>
<tr class="even">
<td align="left">KJ634464 Felis chaus</td>
<td align="left">KJ634464</td>
<td align="left">Felis_chaus</td>
<td align="left">cytochrome c oxidase subunit I.fasta</td>
<td align="left">Felis_chaus</td>
</tr>
<tr class="odd">
<td align="left">ON377297 Vulpes corsac</td>
<td align="left">ON377297</td>
<td align="left">Vulpes_corsac</td>
<td align="left">cytochrome c oxidase subunit I.fasta</td>
<td align="left">Vulpes_corsac</td>
</tr>
<tr class="even">
<td align="left">OM980695 Vulpes vulpes</td>
<td align="left">OM980695</td>
<td align="left">Vulpes_vulpes</td>
<td align="left">cytochrome c oxidase subunit I.fasta</td>
<td align="left">Vulpes_vulpes</td>
</tr>
<tr class="odd">
<td align="left">MT731651 Vulpes lagopus</td>
<td align="left">MT731651</td>
<td align="left">Vulpes_lagopus</td>
<td align="left">cytochrome c oxidase subunit I.fasta</td>
<td align="left">Vulpes_lagopus</td>
</tr>
<tr class="even">
<td align="left">AF028205 Vulpes macrotis</td>
<td align="left">AF028205</td>
<td align="left">Vulpes_macrotis</td>
<td align="left">cytochrome c oxidase subunit I.fasta</td>
<td align="left">Vulpes_macrotis</td>
</tr>
<tr class="odd">
<td align="left">AF028194 Vulpes zerda</td>
<td align="left">AF028194</td>
<td align="left">Vulpes_zerda</td>
<td align="left">cytochrome c oxidase subunit I.fasta</td>
<td align="left">Vulpes_zerda</td>
</tr>
<tr class="even">
<td align="left">GU733705 Phoca vitulina</td>
<td align="left">GU733705</td>
<td align="left">Phoca_vitulina</td>
<td align="left">cytochrome c oxidase subunit I.fasta</td>
<td align="left">Phoca_vitulina</td>
</tr>
<tr class="odd">
<td align="left">OM220069 Manis pentadactyla</td>
<td align="left">OM220069</td>
<td align="left">Manis_pentadactyla</td>
<td align="left">cytochrome c oxidase subunit I.fasta</td>
<td align="left">Manis_pentadactyla</td>
</tr>
<tr class="even">
<td align="left">MN537981 Felis silvestris</td>
<td align="left">MN537981</td>
<td align="left">Felis_silvestris</td>
<td align="left">NADH dehydrogenase subunit 5.fasta</td>
<td align="left">Felis_silvestris</td>
</tr>
<tr class="odd">
<td align="left">MN313781 Felis catus</td>
<td align="left">MN313781</td>
<td align="left">Felis_catus</td>
<td align="left">NADH dehydrogenase subunit 5.fasta</td>
<td align="left">Felis_catus</td>
</tr>
<tr class="even">
<td align="left">MK606132 Felis margarita</td>
<td align="left">MK606132</td>
<td align="left">Felis_margarita</td>
<td align="left">NADH dehydrogenase subunit 5.fasta</td>
<td align="left">Felis_margarita</td>
</tr>
<tr class="odd">
<td align="left">GU561700 Felis chaus</td>
<td align="left">GU561700</td>
<td align="left">Felis_chaus</td>
<td align="left">NADH dehydrogenase subunit 5.fasta</td>
<td align="left">Felis_chaus</td>
</tr>
<tr class="even">
<td align="left">AF006400 Felis nigripes</td>
<td align="left">AF006400</td>
<td align="left">Felis_nigripes</td>
<td align="left">NADH dehydrogenase subunit 5.fasta</td>
<td align="left">Felis_nigripes</td>
</tr>
<tr class="odd">
<td align="left">AH014073 Vulpes lagopus</td>
<td align="left">AH014073</td>
<td align="left">Vulpes_lagopus</td>
<td align="left">NADH dehydrogenase subunit 5.fasta</td>
<td align="left">Vulpes_lagopus</td>
</tr>
<tr class="even">
<td align="left">AY377376 Phoca groenlandica</td>
<td align="left">AY377376</td>
<td align="left">Pagophilus_groenlandicus</td>
<td align="left">NADH dehydrogenase subunit 5.fasta</td>
<td align="left">Phoca_groenlandica</td>
</tr>
<tr class="odd">
<td align="left">GU167331 Phoca fasciata</td>
<td align="left">GU167331</td>
<td align="left">Histriophoca_fasciata</td>
<td align="left">NADH dehydrogenase subunit 5.fasta</td>
<td align="left">Phoca_fasciata</td>
</tr>
</tbody>
</table>
<p>We‚Äôll now align the sequences that we just curated. For this, we just
use <code><a href="../reference/sq.aln.html">sq.aln()</a></code> with default parameters. We need to indicate
that we‚Äôre interested in aligning only the <code>"renamed"</code>
<code>fasta</code> files in our <code>1.CuratedSequences</code>
folder.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/sq.aln.html">sq.aln</a></span><span class="op">(</span>folder <span class="op">=</span> <span class="st">'1.CuratedSequences'</span>, FilePatterns <span class="op">=</span> <span class="st">"renamed"</span><span class="op">)</span></code></pre></div>
<p>The resulting multiple sequence alignments will be saved to the
<code>2.Alignments</code> folder. In that new folder, we will have two
types of files: (1) raw alignments (same file names as in
<code>1.CuratedSequences</code>) and (2) alignments with ambiguous sites
removed (<code>Masked_*</code> prefix). Masked alignments are only
created if the <code>mask</code> argument in <code>sq.aln</code> is set
to <code>TRUE</code>. In that case, one additional <code>.csv</code>
file is created for each of the alignments
(<code>0.Masked.Information_*</code>). Each of these datasets list the
number of sites in the masked alignment that (1) are not gaps
(<code>NonGaps</code> column), (2) if the sequence was removed due to
the elevated number of gaps (<code>removedPerGaps</code>; controlled
using the <code>threshold</code> argument in <code>sq.aln</code>), or
(3) if it was removed directly in the masking step
(<code>removedMasking</code>). Note that, for some gene regions, making
can fail. In that case, only the original alignment file is saved to the
<code>2.Alignments</code> folder.</p>
<p>Note that we could use these resulting alignments directly to infer
our phylogenies. We cover these steps within <code>phruta</code> in
another vignette: ‚ÄúPhylogenetics with the <code>phruta</code> R
package‚Äù. For now, let‚Äôs wrap up and plot one of our (cool) alignments.
Let‚Äôs first check the raw alignments!</p>
<div class="figure">
<img src="EDF.Raw.jpeg" title="Raw alignments" alt=""><p class="caption">A figure showing raw alignments</p>
</div>
<p>Now, the masked alignments‚Ä¶!!</p>
<div class="figure">
<img src="EDF.Cur.jpeg" title="Curated alignments" alt=""><p class="caption">A figure showing curated alignments</p>
</div>
<p>And we‚Äôre done for now!! Thanks for following this tutorial‚Ä¶:)</p>
<p>In total, this vignette took 11 minutes to render in my local
machine. You can now try to run <code>phruta</code> using your favorite
groups organisms! Don‚Äôt forget to check the other tutorials and get in
touch if you find any issues‚Ä¶Buena suerte!</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="http://cromanpa94.github.io/cromanpa/" class="external-link">Cristian Roman Palacios</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
