<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="phruta">
<title>Phylogenetics with phruta • phruta</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Phylogenetics with phruta">
<meta property="og:description" content="phruta">
<meta property="og:image" content="https://cromanpa94.github.io/phruta/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">phruta</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/phruta.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../articles/index.html">Articles</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="http://twitter.com/cromanpa" aria-label="Twitter">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/cromanpa94/phruta/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Phylogenetics with phruta</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/cromanpa94/phruta/blob/HEAD/vignettes/Phylogenetics_phruta.Rmd" class="external-link"><code>vignettes/Phylogenetics_phruta.Rmd</code></a></small>
      <div class="d-none name"><code>Phylogenetics_phruta.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="table-of-contents">Table of Contents<a class="anchor" aria-label="anchor" href="#table-of-contents"></a>
</h2>
<ol style="list-style-type: decimal">
<li><a href="#paragraph1">Installing RAxML</a></li>
<li><a href="#paragraph2">Installing PATHd-8 and treePL</a></li>
<li><a href="#paragraph3">Phylogenetic inference with <code>phruta</code> and <code>RAxML</code></a></li>
<li><a href="#paragraph4">Tree dating in <code>phruta</code></a></li>
</ol>
<div class="section level3">
<h3 id="installing-raxml">Installing RAxML <a name="paragraph1"></a><a class="anchor" aria-label="anchor" href="#installing-raxml"></a>
</h3>
<p>In <code>MacOS</code>, <code>RAxML</code> can be easily installed to the <code>PATH</code> using one of the two lines below in <code>conda</code>:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true"></a><span class="ex">conda</span> install -c bioconda/label/cf201901 raxml </span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true"></a><span class="ex">conda</span> install -c bioconda raxml</span></code></pre></div>
<p>For other <code>OS</code> (Windows, Linux), please follow the instructions listed in the official <code>RAxML</code> <a href="https://cme.h-its.org/exelixis/web/software/raxml/" class="external-link">website</a></p>
<p>Once <code>RAxML</code> has been installed to your computer, open <code>R</code> and make sure that the following line doesn’t throw an error.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/system.html" class="external-link">system</a></span><span class="op">(</span><span class="st">"raxmlHPC"</span><span class="op">)</span></code></pre></div>
<p>Depending on how <code>RAxML</code> was installed, you may want to check if <code>RAxML</code> is called from the terminal using <code>raxmlHPC</code> or <code>raxmlHPC</code>. This string needs to be passed to <code><a href="../reference/tree.raxml.html">tree.raxml()</a></code> using the argument <code>raxml_exec</code>. Please note that this argument corresponds to the <code>exec</code> argument in <code><a href="https://rdrr.io/pkg/ips/man/raxml.html" class="external-link">ips::raxml()</a></code>.</p>
<p>Finally, note that <code>RStudio</code> sometimes has issues finding <em>stuff</em> in the path while using <code><a href="https://rdrr.io/r/base/system.html" class="external-link">system()</a></code>. If you’re using <code>macOS</code>, try starting <code>RStudio</code> from the command line by running the following line:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true"></a><span class="ex">open</span> /Applications/RStudio.app</span></code></pre></div>
<p>In other <code>OS</code>, it might be better to simply avoid using <code>RStudio</code> if you’re interested in running the phylogenetic functions in <code>phruta</code>.</p>
</div>
<div class="section level3">
<h3 id="installing-pathd-8-and-treepl">Installing PATHd-8 and treePL <a name="paragraph2"></a><a class="anchor" aria-label="anchor" href="#installing-pathd-8-and-treepl"></a>
</h3>
<p>There are excellent guides for installing <code>PATHd-8</code> and <code>treePL</code>. Here, I summarize two potentially relevant options.</p>
<p>First, you can use <a href="https://github.com/bomeara/phydocker/blob/master/Dockerfile" class="external-link">Brian O’Meara’s</a> approach for installing <code>PATHd-8</code> in MacOs and linux. I summarize the code in the following <a href="https://gist.github.com/cromanpa94/a43bc710a17220f71d796d6590ea7fe4" class="external-link">link</a>. For Windows users, please use the compiled version of the software provided in the following <a href="https://www2.math.su.se/PATHd8/" class="external-link">link</a>.</p>
<p>Second, you can use homebrew to install <code>treePL</code> (Windows, MacOS, and Linux), thanks to Jonathan Chang.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true"></a><span class="ex">brew</span> install brewsci/bio/treepl</span></code></pre></div>
<p>Please check the following <a href="https://docs.brew.sh/Homebrew-on-Linux" class="external-link">link</a>) if you’re interested in running <code>brew</code> from Windows and Linux.</p>
</div>
<div class="section level3">
<h3 id="phylogenetic-inference-with-phruta-and-raxml">Phylogenetic inference with <code>phruta</code> and <code>RAxML</code> <a name="paragraph3"></a><a class="anchor" aria-label="anchor" href="#phylogenetic-inference-with-phruta-and-raxml"></a>
</h3>
<p>Phylogenetic inference is conducted using the <code><a href="../reference/tree.raxml.html">tree.raxml()</a></code> function. We need to indicate where the aligned sequences are located (<code>folder</code> argument), the patterns of the files in the same folder (<code>FilePatterns</code> argument; “<code>Masked_</code>” in our case). We’ll run a total of 100 boostrap replicates and set the outgroup to “Manis_pentadactyla”.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/tree.raxml.html">tree.raxml</a></span><span class="op">(</span>folder<span class="op">=</span><span class="st">'2.Alignments'</span>, 
           FilePatterns<span class="op">=</span> <span class="st">'Masked_'</span>, 
           raxml_exec<span class="op">=</span><span class="st">'raxmlHPC'</span>, 
           Bootstrap<span class="op">=</span><span class="fl">100</span>, 
           outgroup <span class="op">=</span><span class="st">"Manis_pentadactyla"</span><span class="op">)</span></code></pre></div>
<p>The trees are saved in <code>3.Phylogeny</code>. Likely, the bipartitions tree, “RAxML_bipartitions.phruta”, is the most relevant. <code>3.Phylogeny</code> also includes additional <code>RAxML</code>-related input and output files.</p>
<p>Finally, let’s perform tree dating in our phylogeny using secondary calibrations extracted from <a href="https://royalsocietypublishing.org/doi/pdf/10.1098/rspb.2016.1334" class="external-link">Scholl and Wiens (2016)</a>. This study curated potentially the most comprenhensive and reliable set of trees to summarize the temporal dimension in evolution across the tree of life. In <code>phruta</code>, the trees from Scholl and Wiens (2016) were renamed to match taxonomic groups.</p>
</div>
<div class="section level3">
<h3 id="tree-dating-in-phruta">Tree dating in <code>phruta</code> <a name="paragraph4"></a><a class="anchor" aria-label="anchor" href="#tree-dating-in-phruta"></a>
</h3>
<p>Tree dating is performed using the <code><a href="../reference/tree.dating.html">tree.dating()</a></code> function in <code>phruta</code>. We have to provide the name of the folder containing the <code>1.Taxonomy.csv</code> file created in <code><a href="../reference/sq.curate.html">sq.curate()</a></code>. We also have to indicate the name of the folder containing the <code>RAxML_bipartitions.phruta</code> file. We will scale our phylogeny using <code>treePL</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/tree.dating.html">tree.dating</a></span><span class="op">(</span>taxonomyFolder<span class="op">=</span><span class="st">"1.CuratedSequences"</span>, 
            phylogenyFolder<span class="op">=</span><span class="st">"3.Phylogeny"</span>, 
            scale<span class="op">=</span><span class="st">'treePL'</span><span class="op">)</span></code></pre></div>
<p>Running this line will result in a new folder <code>4.Timetree</code>, including the different time-calibrated phylogenies obained (if any) and associated secondary calibrations used in the analyses. We found only a few overlapping calibration points (family-level constraints):</p>
<p>Here’s the resulting time-calibrated phylogeny. The whole process took ~20 minutes to complete on my computer (16 gb RAM, i5).</p>
<div class="figure">
<img src="tree.png" alt=""><p class="caption">Time-calibrated phylogeny for the target taxa. Analyses performed within the phruta R package.</p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="http://cromanpa94.github.io/cromanpa/" class="external-link">Cristian Roman Palacios</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
