<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="phruta">
<title>phruta behind the scenes • phruta</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="phruta behind the scenes">
<meta property="og:description" content="phruta">
<meta property="og:image" content="https://ropensci.github.io/phruta/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">phruta</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/phruta.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../articles/index.html">Articles</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="http://twitter.com/cromanpa" aria-label="Twitter">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/ropensci/phruta/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="package_acknowledgements_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>phruta behind the scenes</h1>
                        <h4 data-toc-skip class="author">Cristian Román-Palacios</h4>
            
            <h4 data-toc-skip class="date">First created on 2021-07-30. Updated on 2023-05-02</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/ropensci/phruta/blob/HEAD/vignettes/package_acknowledgements.Rmd" class="external-link"><code>vignettes/package_acknowledgements.Rmd</code></a></small>
      <div class="d-none name"><code>package_acknowledgements.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="phruta-behind-the-scenes">
<code>phruta</code> behind the scenes<a class="anchor" aria-label="anchor" href="#phruta-behind-the-scenes"></a>
</h2>
<p><code>phruta</code> has a plethora of dependencies that enables the package to perform particular tasks in a (largely…) organized fashion. This vignette has the objective of acknowledging the maintainers and creators of the main R packages that <code>phruta</code> uses in different of it’s steps. First, both <code>rgbif</code> (Chamberlain and Boettiger, 2017) and <code>taxize</code> (Chamberlain and Szöcs, 2013) are both used for either retrieving taxonomic information or curating it. Second, <code>ape</code> (Paradis and Schliep, 2019) is generally used for interacting with sequence- and tree-based files within <code>R</code> and relative to files in the working directory. Third, both <code>rentrez</code> (Winter, 2017) and <code>reutils</code> (Schöfl, 2016) are used to retrieve sequences from GenBank. We use functions from two packages to interact with GenBank given differences in the structure of search strings or expectations in the resulting objects. Fourth, we use <code>DECIPHER</code> (Wright, 2016), <code>Biostrings</code> (Pagès et al. 2022), <code>msa</code> (Bodenhofer et al. 2015), and <code>odseq</code> (Jiménez, 2022) mostly during sequence alignment-related steps involving both curation and effective alignment. Fifth, we use <code>ips</code> (Heibl, 2008) to interact with external software such as <code>RAxML</code> (Stamatakis, 2014). Finally, <code>geiger</code> (Harmon et al. 2008; Pennell et al. 2014) and <code>Rogue</code> (Aberer et al. 2013; Smith, 2021, 2022) are used to time-calibrate a phylogeny using a reference tree and identify rogue taxa in newly-inferred trees, respectively.</p>
<ul>
<li><p>Aberer, A. J., Krompass, D., &amp; Stamatakis, A. (2013). Pruning rogue taxa improves phylogenetic accuracy: an efficient algorithm and webservice. Systematic biology, 62(1), 162-166.</p></li>
<li><p>Bodenhofer, U., Bonatesta, E., Horejš-Kainrath, C., &amp; Hochreiter, S. (2015). msa: an R package for multiple sequence alignment. Bioinformatics, 31(24), 3997-3999.</p></li>
<li><p>Chamberlain, S. A., &amp; Boettiger, C. (2017). R Python, and Ruby clients for GBIF species occurrence data (No. e3304v1). PeerJ Preprints.</p></li>
<li><p>Chamberlain, S. A., &amp; Szöcs, E. (2013). taxize: taxonomic search and retrieval in R. F1000Research, 2. <a href="doi:10.5281/zenodo.5037327" class="uri">doi:10.5281/zenodo.5037327</a></p></li>
<li><p>Harmon, L. J., Weir, J. T., Brock, C. D., Glor, R. E., &amp; Challenger, W. (2008). GEIGER: investigating evolutionary radiations. Bioinformatics, 24(1), 129-131.</p></li>
<li><p>Heibl, C. (2008). PHYLOCH: R language tree plotting tools and interfaces to d Jiménez, J. (2022). odseq: Outlier detection in multiple sequence alignments. R package version 1.24.0.</p></li>
<li><p>Pagès, H., Aboyoun, P., Gentleman, R., DebRoy, S. (2022). Biostrings: Efficient manipulation of biological strings. R package version 2.64.0</p></li>
<li><p>Paradis, E., &amp; Schliep, K. (2019). ape 5.0: an environment for modern phylogenetics and evolutionary analyses in R. Bioinformatics, 35(3), 526-528.</p></li>
<li><p>Pennell, M. W., Eastman, J. M., Slater, G. J., Brown, J. W., Uyeda, J. C., FitzJohn, R. G., … &amp; Harmon, L. J. (2014). geiger v2. 0: an expanded suite of methods for fitting macroevolutionary models to phylogenetic trees. Bioinformatics, 30(15), 2216-2218.</p></li>
<li><p>Schöfl, G. (2016). reutils: Talk to the NCBI EUtils. R package version 0.2.3.</p></li>
<li><p>Smith, M. R. (2022). Using information theory to detect rogue taxa and improve consensus trees. Systematic Biology, 71(5), 1088-1094.</p></li>
<li><p>Smith, M.R. (2021): Rogue: Identify Rogue Taxa in Sets of Phylogenetic Trees, Zotero.</p></li>
<li><p>Stamatakis, A. (2014). RAxML version 8: a tool for phylogenetic analysis and post-analysis of large phylogenies. Bioinformatics, 30(9), 1312-1313.</p></li>
<li><p>Winter, D. J. (2017). rentrez: An R package for the NCBI eUtils API (No. e3179v2). PeerJ Preprints.</p></li>
<li><p>Wright, E. S. (2016). Using DECIPHER v2. 0 to analyze big biological sequence data in R. R J., 8(1), 352.</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="general-acknowledgements">General acknowledgements<a class="anchor" aria-label="anchor" href="#general-acknowledgements"></a>
</h2>
<p>The author thanks Heidi E. Steiner for proofreading the vignettes and documentation in <code>phruta</code> in addition to early versions of this manuscript. Anna Krystalli, Rayna Harris, Frederick Boehm, and Maëlle Salmon provided excellent comments during the peer review process of <code>phruta</code> in ROpenSci. Wonkyiun Yim provided early support in <code>salphycon</code>. Finally, the author thanks the author and maintainers of all the packages used in <code>phruta</code> and <code>salphycon</code> for their constant support to the field.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="http://ropensci.github.io/cromanpa/" class="external-link">Cristian Roman Palacios</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
