<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="phruta">
<title>Using phruta with a defined set of target genes • phruta</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Using phruta with a defined set of target genes">
<meta property="og:description" content="phruta">
<meta property="og:image" content="https://cromanpa94.github.io/phruta/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">phruta</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/phruta.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../articles/index.html">Articles</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="http://twitter.com/cromanpa" aria-label="Twitter">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/cromanpa94/phruta/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="phruta_targetgenes_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Using phruta with a defined set of target genes</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/cromanpa94/phruta/blob/HEAD/vignettes/phruta_targetgenes.Rmd" class="external-link"><code>vignettes/phruta_targetgenes.Rmd</code></a></small>
      <div class="d-none name"><code>phruta_targetgenes.Rmd</code></div>
    </div>

    
    
<p>In this vignette, we will be retrieving sequences from genbank using <code>phruta</code>. We will need a list of taxa and the names of particular gene regions that need to be sampled. If you have no idea on what genes you should sample - <code>phruta</code>’ve got your back. Just check the “Introduction to the `phruta` R package” and “To export or not export `phruta` outputs” vignettes.</p>
<p>Now, given that we already have some target genes in mind, we can avoid using <code><a href="../reference/gene.sampling.retrieve.html">gene.sampling.retrieve()</a></code>. We won’t generate an accession table using <code><a href="../reference/acc.table.retrieve.html">acc.table.retrieve()</a></code> at the beginning, mostly because the sampling that we’re using in <code>phruta</code> this time. Therefore, <code><a href="../reference/sq.retrieve.direct.html">sq.retrieve.direct()</a></code> is less flexible than <code><a href="../reference/sq.retrieve.indirect.html">sq.retrieve.indirect()</a></code>. As noted before, <code><a href="../reference/sq.retrieve.direct.html">sq.retrieve.direct()</a></code> does its best to directly (i.e. without input from the user) retrieve sequences for a target set of taxa and set of gene regions. You are more likely to catch errors using <code><a href="../reference/sq.retrieve.indirect.html">sq.retrieve.indirect()</a></code>. However, mistakes will be harder to spot and fix when using <code><a href="../reference/sq.retrieve.direct.html">sq.retrieve.direct()</a></code>.</p>
<p>Let’s first start by loading <code>phruta</code>!</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/cromanpa94/phruta" class="external-link">phruta</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        ✅  Thank you for using the {phruta} R package!   ✅</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##                         🙂Happy coding!!🙂</span></span>
<span><span class="co">##                        🍊🍍🍐🥝🥥🌶🥑🍒🥭</span></span></code></pre>
<p>Let’s download gene sequences for the taxa in <em>Felis</em>, <em>Vulpes</em>, and <em>Phoca</em>. Tjhis species-level sampling corresponds with the one in other vignettes. We will be sampling only two gene regions using <code><a href="../reference/sq.retrieve.direct.html">sq.retrieve.direct()</a></code>. Note that we’re going to be using the list of species for the target genera from the gbif taxonomic backbone (hence <code>db = "gbif"</code>). There are more databases available for this (please check <code><a href="https://docs.ropensci.org/taxize/reference/downstream.html" class="external-link">taxize::downstream()</a></code>).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sq.retrieve.direct.html">sq.retrieve.direct</a></span><span class="op">(</span></span>
<span>  clades <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Felis"</span>, <span class="st">"Vulpes"</span>, <span class="st">"Phoca"</span><span class="op">)</span>,</span>
<span>  species <span class="op">=</span> <span class="st">"Manis_pentadactyla"</span>,</span>
<span>  genes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ADORA3"</span>, <span class="st">"CYTB"</span><span class="op">)</span>,</span>
<span>  db <span class="op">=</span> <span class="st">"gbif"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The code above will create a folder <code>0.Sequences</code> in your working directory. Note that <code><a href="../reference/sq.retrieve.direct.html">sq.retrieve.direct()</a></code> does not return objects to the environment. The rest of this tutorial follow the same structure as the “To export or not export `phruta` outputs” vignette.</p>
<p>For instance, we can use the <code><a href="../reference/sq.curate.html">sq.curate()</a></code> function to remove highly divergent sequences. This function will also remove observations from species that are not within our target taxonomic groups.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sq.curate.html">sq.curate</a></span><span class="op">(</span>filterTaxonomicCriteria <span class="op">=</span> <span class="st">'Felis|Vulpes|Phoca|Manis'</span>,</span>
<span>                         kingdom <span class="op">=</span> <span class="st">'animals'</span>, </span>
<span>                         folder <span class="op">=</span> <span class="st">"0.Sequences"</span>,</span>
<span>                         removeOutliers <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                         minSeqs <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p>Let’s review the current sampling…</p>
<table class="table">
<caption>4. Preliminary accession number table</caption>
<colgroup>
<col width="79%">
<col width="3%">
<col width="6%">
<col width="4%">
<col width="6%">
</colgroup>
<thead><tr class="header">
<th align="left">OriginalNames</th>
<th align="left">AccN</th>
<th align="left">Species</th>
<th align="left">file</th>
<th align="left">OldSpecies</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">AY011246.1 Felis catus adenosine A3 receptor (ADORA3) gene, partial cds</td>
<td align="left">AY011246</td>
<td align="left">Felis_catus</td>
<td align="left">ADORA3.fasta</td>
<td align="left">Felis_catus</td>
</tr>
<tr class="even">
<td align="left">GU167614.1 Vulpes lagopus adenosine A3 receptor (ADORA3) gene, exon 2 and partial cds</td>
<td align="left">GU167614</td>
<td align="left">Vulpes_lagopus</td>
<td align="left">ADORA3.fasta</td>
<td align="left">Vulpes_lagopus</td>
</tr>
<tr class="odd">
<td align="left">GU167677.1 Phoca largha adenosine A3 receptor (ADORA3) gene, exon 2 and partial cds</td>
<td align="left">GU167677</td>
<td align="left">Phoca_largha</td>
<td align="left">ADORA3.fasta</td>
<td align="left">Phoca_largha</td>
</tr>
<tr class="even">
<td align="left">GU930874.1 Phoca vitulina adenosine A3 receptor (ADORA3) gene, partial cds</td>
<td align="left">GU930874</td>
<td align="left">Phoca_vitulina</td>
<td align="left">ADORA3.fasta</td>
<td align="left">Phoca_vitulina</td>
</tr>
<tr class="odd">
<td align="left">AY011251.1 Manis pentadactyla adenosine A3 receptor (ADORA3) gene, partial cds</td>
<td align="left">AY011251</td>
<td align="left">Manis_pentadactyla</td>
<td align="left">ADORA3.fasta</td>
<td align="left">Manis_pentadactyla</td>
</tr>
<tr class="even">
<td align="left">MH558234.1 Felis silvestris bieti isolate FBIPHap2 NADH dehydrogenase subunit 5 (ND5) and NADH dehydrogenase subunit 6 (ND6) genes, complete cds; tRNA-Glu gene, complete sequence; and cytochrome b (CytB) gene, partial cds; mitochondrial</td>
<td align="left">MH558234</td>
<td align="left">Felis_silvestris</td>
<td align="left">CYTB.fasta</td>
<td align="left">Felis_silvestris</td>
</tr>
<tr class="odd">
<td align="left">MW267829.1 Felis catus isolate VA79_2017 cytochrome b (cytb) gene, partial cds; mitochondrial</td>
<td align="left">MW267829</td>
<td align="left">Felis_catus</td>
<td align="left">CYTB.fasta</td>
<td align="left">Felis_catus</td>
</tr>
<tr class="even">
<td align="left">MN370575.1 Felis chaus isolate Jungle Cat 5 cytochrome b (cytb) gene, partial cds; mitochondrial</td>
<td align="left">MN370575</td>
<td align="left">Felis_chaus</td>
<td align="left">CYTB.fasta</td>
<td align="left">Felis_chaus</td>
</tr>
<tr class="odd">
<td align="left">OM970983.1 Otocolobus manul isolate 6-1-11B cytochrome b (cytb) gene, partial cds; mitochondrial</td>
<td align="left">OM970983</td>
<td align="left">Felis_manul</td>
<td align="left">CYTB.fasta</td>
<td align="left">Otocolobus_manul</td>
</tr>
<tr class="even">
<td align="left">MK606132.1 Felis margarita haplotype AH NADH dehydrogenase subunit 5 (ND5) and cytochrome b (cytb) genes, partial cds; and tRNA-Thr gene and D-loop, partial sequence; mitochondrial</td>
<td align="left">MK606132</td>
<td align="left">Felis_margarita</td>
<td align="left">CYTB.fasta</td>
<td align="left">Felis_margarita</td>
</tr>
<tr class="odd">
<td align="left">KU378587.1 Vulpes cana isolate B.F.Y3 cytochrome b (cytb) gene, partial cds; mitochondrial</td>
<td align="left">KU378587</td>
<td align="left">Vulpes_cana</td>
<td align="left">CYTB.fasta</td>
<td align="left">Vulpes_cana</td>
</tr>
<tr class="even">
<td align="left">MT795179.1 Vulpes corsac isolate SH21 cytochrome b (CYTB) gene, complete cds; mitochondrial</td>
<td align="left">MT795179</td>
<td align="left">Vulpes_corsac</td>
<td align="left">CYTB.fasta</td>
<td align="left">Vulpes_corsac</td>
</tr>
<tr class="odd">
<td align="left">EU872065.1 Vulpes ferrilata haplotype 1 cytochrome b (cytb) gene, partial cds; mitochondrial</td>
<td align="left">EU872065</td>
<td align="left">Vulpes_ferrilata</td>
<td align="left">CYTB.fasta</td>
<td align="left">Vulpes_ferrilata</td>
</tr>
<tr class="even">
<td align="left">KX093945.1 Vulpes lagopus haplotype 5 cytochrome b (cytb) gene, partial cds; mitochondrial</td>
<td align="left">KX093945</td>
<td align="left">Vulpes_lagopus</td>
<td align="left">CYTB.fasta</td>
<td align="left">Vulpes_lagopus</td>
</tr>
<tr class="odd">
<td align="left">AF028157.1 Vulpes macrotis cytochrome b (cytb) gene, mitochondrial gene encoding mitochondrial protein, partial cds</td>
<td align="left">AF028157</td>
<td align="left">Vulpes_macrotis</td>
<td align="left">CYTB.fasta</td>
<td align="left">Vulpes_macrotis</td>
</tr>
<tr class="even">
<td align="left">KJ597964.1 Vulpes pallida haplotype PMa2 cytochrome b (cytb) gene, partial cds; mitochondrial</td>
<td align="left">KJ597964</td>
<td align="left">Vulpes_pallida</td>
<td align="left">CYTB.fasta</td>
<td align="left">Vulpes_pallida</td>
</tr>
<tr class="odd">
<td align="left">KU378373.1 Vulpes rueppellii isolate R.F.Y6 cytochrome b (cytb) gene, partial cds; mitochondrial</td>
<td align="left">KU378373</td>
<td align="left">Vulpes_rueppellii</td>
<td align="left">CYTB.fasta</td>
<td align="left">Vulpes_rueppellii</td>
</tr>
<tr class="even">
<td align="left">MK244494.1 Vulpes vulpes isolate LD124 haplotype FOX cytochrome b (CYTB) gene, partial cds; mitochondrial</td>
<td align="left">MK244494</td>
<td align="left">Vulpes_vulpes</td>
<td align="left">CYTB.fasta</td>
<td align="left">Vulpes_vulpes</td>
</tr>
<tr class="odd">
<td align="left">MH854561.1 Vulpes zerda isolate X161349 cytochrome b (Cytb) gene, partial cds; mitochondrial</td>
<td align="left">MH854561</td>
<td align="left">Vulpes_zerda</td>
<td align="left">CYTB.fasta</td>
<td align="left">Vulpes_zerda</td>
</tr>
<tr class="even">
<td align="left">LC466150.1 Phoca largha PLCBNo6 mitochondrial cytb gene for cytochrome b, partial cds</td>
<td align="left">LC466150</td>
<td align="left">Phoca_largha</td>
<td align="left">CYTB.fasta</td>
<td align="left">Phoca_largha</td>
</tr>
<tr class="odd">
<td align="left">AB510422.1 Phoca vitulina stejnegeri mitochondrial cytb gene for cytochrome b, complete cds, isolate: Pvs15</td>
<td align="left">AB510422</td>
<td align="left">Phoca_vitulina</td>
<td align="left">CYTB.fasta</td>
<td align="left">Phoca_vitulina</td>
</tr>
<tr class="even">
<td align="left">OP748355.1 Manis pentadactyla voucher MBSV033 cytochrome b (cytb) gene, partial cds; mitochondrial</td>
<td align="left">OP748355</td>
<td align="left">Manis_pentadactyla</td>
<td align="left">CYTB.fasta</td>
<td align="left">Manis_pentadactyla</td>
</tr>
</tbody>
</table>
<p>Now, we’ll align the sequences that we just curated. For this, we use <code><a href="../reference/sq.aln.html">sq.aln()</a></code> with under default parameters.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/sq.aln.html">sq.aln</a></span><span class="op">(</span>folder <span class="op">=</span> <span class="st">'1.CuratedSequences'</span>, FilePatterns <span class="op">=</span> <span class="st">"renamed"</span><span class="op">)</span></span></code></pre></div>
<p>The resulting multiple sequence alignments will be saved to <code>sqs.aln()</code> object, a list. For each of the gene regions, we will have access to the original alignment (<code>Aln.Original</code>), the masked one (<code>Aln.Masked</code>), and information on the masking process. Let’s first check the raw alignments…</p>
<p><img src="PTG.Raw.jpeg" title="Raw alignments" alt="A figure showing raw alignments"> Now, the masked alignments…</p>
<div class="figure">
<img src="PTG.Cur.jpeg" title="Curated alignments" alt=""><p class="caption">A figure showing curated alignments</p>
</div>
<p>And we’re done for now!! You can compare the sampling between the main three tutorials using <code><a href="../reference/sq.retrieve.direct.html">sq.retrieve.direct()</a></code> and <code><a href="../reference/sq.retrieve.indirect.html">sq.retrieve.indirect()</a></code>. In total, this vignette took 3 minutes to render in my local machine.</p>
  </main>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="http://cromanpa94.github.io/cromanpa/" class="external-link">Cristian Roman Palacios</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
